// @aurum/pfe-ui/es/common/utils/dictUtil.js
import { queryAllDict } from '../api/dictAPI';
import { setLocalItem } from './cache';
import { isEmpty } from 'lodash';
/**
 * 获取字典数据列表
 * @param type
 * @returns {*}
 */

export function getDictionary(type) {
  return new Promise(function (resolve, reject) {
    queryAllDict().then(function (response) {
      var data = response.data || [];
      var dataMapping = {};
      var dataListMapping = {};
      data.forEach(function (item) {
        var itemData = [];
        item.dictDataList.forEach(function (i) {
          loopDictData(i, item.dictType, dataMapping, itemData);
        });
        dataListMapping[item.dictType] = itemData;
      });
      setLocalItem('DICTIONARY', JSON.stringify(dataMapping), '');
      setLocalItem('DICTIONARYLIST', JSON.stringify(dataListMapping), '');
      var list = {};

      if (type === 'DICTIONARY') {
        list = dataMapping;
      } else {
        list = dataListMapping;
      }

      resolve(list);
    });
  });
}
;
export function loopDictData(data, dictType, dataMapping, itemData) {
  var tempData = {};
  tempData = {
    dictLabelCn: data.dictLabelCn,
    dictLabelEn: data.dictLabelEn,
    dictValue: data.dictValue
  };

  if (data.status === '1') {
    dataMapping["".concat(dictType).concat(data.dictValue)] = tempData;
    itemData.push(tempData);
  }

  if (!isEmpty(data.children)) {
    data.children.forEach(function (item) {
      loopDictData(item, dictType, dataMapping, itemData);
    });
  }
}
;
