// @aurum/pfe-ui/es/tree-select/TreeSelect.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { TreeSelect as AntdTreeSelect, ConfigProvider } from 'antd';
import { IconDownFill, IconOpenB } from '@aurum/icons';
var TreeSelect = function TreeSelect(props) {
  var _React$useContext = React.useContext(ConfigProvider.ConfigContext),
    getPrefixCls = _React$useContext.getPrefixCls;
  return /*#__PURE__*/React.createElement(AntdTreeSelect, _extends({
    switcherIcon: /*#__PURE__*/React.createElement("span", {
      className: "".concat(getPrefixCls(), "-switcher-icon")
    }, /*#__PURE__*/React.createElement(IconDownFill, {
      style: {
        fontSize: '12px'
      }
    })),
    suffixIcon: /*#__PURE__*/React.createElement(IconOpenB, {
      style: {
        strokeWidth: 0
      }
    })
  }, props));
};
TreeSelect.TreeNode = AntdTreeSelect.TreeNode;
TreeSelect.SHOW_ALL = AntdTreeSelect.SHOW_ALL;
TreeSelect.SHOW_PARENT = AntdTreeSelect.SHOW_PARENT;
TreeSelect.SHOW_CHILD = AntdTreeSelect.SHOW_CHILD;
export default TreeSelect;
