// @aurum/pfe-ui/es/navigation-bar/NavigationBar.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import { IconOpenB } from '@aurum/icons';
import React from 'react';
import { Divider, Dropdown, Layout } from '../index';

var NavigationBar = /*#__PURE__*/function (_React$Component) {
  _inherits(NavigationBar, _React$Component);

  var _super = _createSuper(NavigationBar);

  function NavigationBar() {
    _classCallCheck(this, NavigationBar);

    return _super.apply(this, arguments);
  }

  _createClass(NavigationBar, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          toggleIcon = _this$props.toggleIcon,
          tabContent = _this$props.tabContent,
          settingButtons = _this$props.settingButtons,
          dropdownMenu = _this$props.dropdownMenu,
          avatar = _this$props.avatar,
          _this$props$userName = _this$props.userName,
          userName = _this$props$userName === void 0 ? '' : _this$props$userName;
      return /*#__PURE__*/React.createElement(Layout, null, /*#__PURE__*/React.createElement(Layout.Header, null, /*#__PURE__*/React.createElement("div", {
        className: "navigation-bar"
      }, /*#__PURE__*/React.createElement("div", {
        className: "left-part"
      }, toggleIcon()), tabContent(), /*#__PURE__*/React.createElement("div", {
        className: "right-part"
      }, settingButtons(), /*#__PURE__*/React.createElement(Divider, {
        type: "vertical",
        style: {
          height: 16,
          marginRight: '24px',
          top: 0
        }
      }), /*#__PURE__*/React.createElement("div", {
        className: "user-button"
      }, /*#__PURE__*/React.createElement(Dropdown, {
        overlay: dropdownMenu,
        placement: "bottomRight"
      }, /*#__PURE__*/React.createElement("div", {
        className: "user-wrapper"
      }, avatar(), /*#__PURE__*/React.createElement("div", {
        className: "user-name"
      }, userName), /*#__PURE__*/React.createElement(IconOpenB, {
        style: {
          fontSize: "12px"
        }
      }))))))));
    }
  }]);

  return NavigationBar;
}(React.Component);

export default NavigationBar;
