// @omc/boss-common/dist/utils/common.js
import { getLocalItem, setLocalItem, Constants } from './cache';
import { getAllRegion, getALlDictionaryDataList } from '../api/sysAPI';
export function getDictList(type, data) {
  var options = [];

  var _loop = function _loop(key) {
    if (data.hasOwnProperty(key) && key === type) {
      data[key].map(function (item) {
        var dictLabel = checkLocaleIsCn() ? item.dictLabelCn : item.dictLabelEn;
        options.push({
          label: dictLabel,
          value: item.dictValue,
          key: key + dictLabel
        });
      });
    }
  };

  for (var key in data) {
    _loop(key);
  }

  return options;
}
export function checkMyPermission(code) {
  var data = window.localStorage.getItem(Constants.MYPERMISSION);
  var userInfo = getLocalItem(Constants.USER_INFO);

  if (data && userInfo) {
    if (userInfo.username === 'admin') {
      return true;
    }

    return data.indexOf(code) !== -1;
  } else {
    return false;
  }
}
export function checkLocaleIsCn() {
  return getLocalItem(Constants.LOCALE) === 'cn';
} // 获取系统参数

export function getValueByKey(key) {
  var value;
  var data = getLocalItem(Constants.PARAMETERS);

  if (data) {
    var parameter = data.find(function (item) {
      return item.parameterKey === key;
    });
    value = parameter && parameter.parameterValue;
  }

  return value;
} // 获取url中的所有参数

export function getAllQueryParams() {
  var str = window.location.href;

  if (!str) {
    return {};
  }

  var num = str.indexOf('?');
  str = str.substr(num + 1); //取得所有参数

  var res = {};
  var name;
  var value;
  var arr = str.split('&'); //各个参数放到数组里

  for (var i = 0; i < arr.length; i++) {
    num = arr[i].indexOf('=');

    if (num > 0) {
      name = arr[i].substring(0, num).trim();
      value = arr[i].substr(num + 1).trim();
      res[name] = value;
    }
  }

  return res;
}

function getCityName(cityInfo, code) {
  var res;
  Object.keys(cityInfo).find(function (k) {
    var key = Object.keys(cityInfo[k]).find(function (ke) {
      return code.toString() === ke;
    });

    if (key) {
      res = cityInfo[k][key];
    }
  });
  return res;
} // 通过code获取城市名称


export function getCityNameByCode(code) {
  var name;
  var cityInfo = getLocalItem('provinceSelect');

  if (cityInfo) {
    name = getCityName(cityInfo, code);
  } else {
    getAllRegion().then(function (res) {
      if (res.code == 2000) {
        setLocalItem('provinceSelect', res.data);
        name = getCityName(res.data, code);
      }
    });
  }

  return name;
} // 字典

export function getDictionary(cb) {
  getALlDictionaryDataList().then(function (response) {
    var data = response.data || [];
    var dataMapping = {};
    var dataListMapping = {};
    data.forEach(function (v) {
      var dicData = {
        dictLabelCn: v.dictLabelCn,
        dictLabelEn: v.dictLabelEn,
        dictValue: v.dictValue
      };
      dataMapping[v.dictType + v.dictValue] = dicData; // 根据字典类型返回列表

      if (v.status === '1') {
        dataListMapping[v.dictType] ? dataListMapping[v.dictType].push(dicData) : dataListMapping[v.dictType] = [dicData];
      }
    });
    setLocalItem(Constants.DICTIONARY, JSON.stringify(dataMapping));
    setLocalItem(Constants.DICTIONARYLIST, JSON.stringify(dataListMapping));

    if (typeof cb === 'function') {
      cb();
    }
  });
}
