// @aurum/pfe-ui/es/common/http.js
import request from './request';
var BASE_API = '/api/inner/boss-api';
var sendHttpRequest = function sendHttpRequest(requests, showLoading) {
  return new Promise(function (resolve, reject) {
    requests.then(function (response) {
      if (response.code === '2000' || response.code === 'SUCCESS') {
        resolve(response); // 返回成功
      } else {
        reject(response);
      }
    })["catch"](function (e) {
      reject(e);
    });
  });
};
export var post = function post(url, param) {
  var prefixUrl = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : BASE_API;
  var showLoading = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : true;
  return sendHttpRequest(request({
    url: prefixUrl + url,
    method: 'post',
    data: param ? param : ''
  }), showLoading);
};
export var put = function put(url, param) {
  var prefixUrl = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : BASE_API;
  var showLoading = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : true;
  return sendHttpRequest(request({
    url: prefixUrl + url,
    method: 'put',
    data: param ? param : ''
  }), showLoading);
};
export var deletes = function deletes(url, param) {
  var prefixUrl = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : BASE_API;
  var showLoading = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : true;
  return sendHttpRequest(request({
    url: prefixUrl + url,
    method: 'delete',
    data: param ? param : ''
  }), showLoading);
};
export var get = function get(url, param) {
  var prefixUrl = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : BASE_API;
  var showLoading = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : true;
  return sendHttpRequest(request({
    url: prefixUrl + url,
    method: 'get',
    params: param ? param : ''
  }), showLoading);
};
