// @aurum/pfe-ui/es/pro-cascader-province/ProCascaderProvince.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { default as Cascader } from '../cascader';
import { getAllRegion } from '../common/api/dictAPI';
var ProCascaderProvince = /*#__PURE__*/function (_React$Component) {
  _inherits(ProCascaderProvince, _React$Component);
  var _super = _createSuper(ProCascaderProvince);
  function ProCascaderProvince() {
    var _this;
    _classCallCheck(this, ProCascaderProvince);
    _this = _super.apply(this, arguments);
    _this.state = {
      options: []
    };
    _this.setStorage = function () {
      var province = localStorage.getItem('province');
      if (province) {
        _this.setState({
          options: JSON.parse(province)
        });
        return true;
      }
      return false;
    };
    _this.fetchData = function () {
      getAllRegion().then(function (res) {
        if (res.code === '2000') {
          var cityData = _this.adapterData(res.data, res.data['000000']);
          _this.setState({
            options: cityData
          });
          localStorage.setItem('province', JSON.stringify(cityData));
        }
      })["catch"](function (err) {
        return console.log(err.msg);
      });
    };
    _this.adapterData = function (arg, arg1) {
      return Object.keys(arg1).map(function (key) {
        if (arg[key]) {
          return {
            label: arg1[key] || '',
            value: key,
            children: _this.adapterData(arg, arg[key])
          };
        } else {
          return {
            label: arg1[key] || '',
            value: key
          };
        }
      });
    };
    return _this;
  }
  _createClass(ProCascaderProvince, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var isLocal = this.setStorage();
      if (!isLocal) {
        this.fetchData();
      }
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(Cascader, _extends({
        options: this.state.options || []
      }, this.props));
    }
  }]);
  return ProCascaderProvince;
}(React.Component);
export default ProCascaderProvince;
