// @mcd/portal-components/dist/components/time-range-section/index.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

import React from "react";
import "./index.less";
import { DatePicker, TimePicker, Row, Col } from "antd";
import moment from "moment";
import ErrorTip from "../error-tips";
import * as DateTimeFormat from "../../utils/format";
var RangePicker = TimePicker.RangePicker;
/**
 *
 *  促销时间范围
 *
 */

var TimeRangeSection = /*#__PURE__*/function (_React$Component) {
  _inherits(TimeRangeSection, _React$Component);

  var _super = _createSuper(TimeRangeSection);

  function TimeRangeSection(props) {
    var _this;

    _classCallCheck(this, TimeRangeSection);

    _this = _super.call(this, props);

    _defineProperty(_assertThisInitialized(_this), "checkStartDate", function () {
      var saveModel = _this.state.saveModel;

      _this.setState({
        startDateTip: saveModel.startDate == "" ? $t("promotion_input_start_date_tip") : ""
      });

      return saveModel.startDate == "" ? false : true;
    });

    _defineProperty(_assertThisInitialized(_this), "checkStartTime", function () {
      var saveModel = _this.state.saveModel;

      _this.setState({
        startTimeTip: saveModel.startTime == "" ? $t("promotion_input_start_time_tip") : ""
      });

      return saveModel.startTime == "" ? false : true;
    });

    _defineProperty(_assertThisInitialized(_this), "checkEndDate", function () {
      var saveModel = _this.state.saveModel;

      _this.setState({
        endDateTip: saveModel.endDate == "" ? $t("promotion_input_end_date_tip") : ""
      });

      return saveModel.endDate == "" ? false : true;
    });

    _defineProperty(_assertThisInitialized(_this), "checkEndTime", function () {
      var endTime = _this.state.endTime;

      _this.setState({
        endTimeTip: endTime == "" ? $t("promotion_input_end_time_tip") : ""
      });

      return endTime == "" ? false : true;
    });

    _defineProperty(_assertThisInitialized(_this), "checkStartEquEnd", function () {
      var _this$state = _this.state,
          saveModel = _this$state.saveModel,
          errorTip = _this$state.errorTip;
      var start = DateTimeFormat.getTimestamp(saveModel.startDate + " " + saveModel.startTime);
      var end = DateTimeFormat.getTimestamp(saveModel.endDate + " " + saveModel.endTime);

      _this.setState({
        endTimeTip: start == end && start != "" ? $t("promotion_dup_start_end_time_tip") : "",
        startTimeTip: start == end && start != "" ? $t("promotion_dup_start_end_time_tip") : "",
        endDateTip: start == end && start != "" ? $t("promotion_dup_start_end_time_tip") : "",
        startDateTip: start == end && start != "" ? $t("promotion_dup_start_end_time_tip") : ""
      });

      return start == end && start != "" ? false : true;
    });

    _defineProperty(_assertThisInitialized(_this), "checkTimeRange", function () {
      var saveModel = _this.state.saveModel;
      var startD = new Promise(function (resolve, reject) {
        var sd = DateTimeFormat.dateTimeToString(saveModel.startDate, DateTimeFormat.dateFormat_);
        _this.checkStartDate() ? resolve(sd) : reject();
      });
      var startT = new Promise(function (resolve, reject) {
        var st = DateTimeFormat.dateTimeToString(saveModel.startTime, DateTimeFormat.timeFormat);
        _this.checkStartTime() ? resolve(st) : reject();
      });
      var endD = new Promise(function (resolve, reject) {
        var ed = DateTimeFormat.dateTimeToString(saveModel.endDate, DateTimeFormat.dateFormat_);
        _this.checkEndDate() ? resolve(ed) : reject();
      });
      var endT = new Promise(function (resolve, reject) {
        var et = DateTimeFormat.dateTimeToString(saveModel.endTime, DateTimeFormat.timeFormat);
        _this.checkEndTime() ? resolve(et) : reject();
      });
      var equ = new Promise(function (resolve, reject) {
        var et = DateTimeFormat.dateTimeToString(saveModel.endTime, DateTimeFormat.timeFormat);
        _this.checkStartEquEnd() ? resolve(et) : reject();
      });
      return Promise.all([startD, startT, endD, endT, equ])["catch"](function (e) {});
    });

    _defineProperty(_assertThisInitialized(_this), "startDateChange", function (value, dateString) {
      var saveModel = _this.state.saveModel;
      var startDateTip = dateString == "" ? _this.state.errorTip : "";
      var start = moment(dateString + " " + saveModel.startTime).format(DateTimeFormat.dateAndTime_);
      saveModel.startDate = dateString.replace(/\//g, "-");

      _this.setState({
        saveModel: _objectSpread({}, saveModel),
        startDateTip: startDateTip,
        resultStartDate: start
      });
    });

    _defineProperty(_assertThisInitialized(_this), "startTimeChange", function (value, dateString) {
      var saveModel = _this.state.saveModel;
      var start = moment(saveModel.startDate + " " + dateString).format(DateTimeFormat.dateAndTime_);
      var startTimeTip = dateString == "" ? _this.state.errorTip : "";
      saveModel.startTime = dateString;

      _this.setState({
        resultStartDate: start,
        saveModel: _objectSpread({}, saveModel),
        startTimeTip: startTimeTip
      });
    });

    _defineProperty(_assertThisInitialized(_this), "endDateChange", function (value, dateString) {
      var saveModel = _this.state.saveModel;
      var endDateTip = dateString == "" ? _this.state.errorTip : "";
      var end = moment(dateString + " " + saveModel.endTime).format(DateTimeFormat.dateAndTime_);
      saveModel.endDate = dateString.replace(/\//g, "-");

      _this.setState({
        resultEndDate: end,
        saveModel: _objectSpread({}, saveModel),
        endDateTip: endDateTip
      });
    });

    _defineProperty(_assertThisInitialized(_this), "endTimeChange", function (value, dateString) {
      var saveModel = _this.state.saveModel;
      var end = moment(saveModel.endDate + " " + dateString).format(DateTimeFormat.dateAndTime_);
      var endTimeTip = dateString == "" ? _this.state.errorTip : "";
      saveModel.endTime = dateString;

      _this.setState({
        resultEndDate: end,
        saveModel: _objectSpread({}, saveModel),
        endTimeTip: endTimeTip
      });
    });

    _this.props.bind && _this.props.bind(_assertThisInitialized(_this));
    _this.state = {
      initialVal: {},
      saveModel: {
        startDate: "",
        startTime: "",
        endDate: "",
        endTime: ""
      },
      errorTip: $t("promotion_select_tip"),
      startDateTip: "",
      endDateTip: "",
      startTimeTip: "",
      endTimeTip: "",
      resultStartDate: "",
      resultEndDate: "",
      startDateLabel: props.startDateLabel || $t("promotion_start_date"),
      endDateLabel: props.endDateLabel || $t("promotion_end_date"),
      startTimeLabel: props.startTimeLabel || $t("promotion_start_time"),
      endTimeLabel: props.endTimeLabel || $t("promotion_end_time")
    };

    if (_this.props.saveModel) {
      _this.state.startDate = _this.props.saveModel.startTime;
      _this.state.startTime = _this.props.saveModel.startTime;
      _this.state.endDate = _this.props.saveModel.endDate;
      _this.state.endTime = _this.props.saveModel.endTime;
    }

    return _this;
  }
  /**
   * 校验开始日期
   */


  _createClass(TimeRangeSection, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      console.log(nextProps, "nextProps render");

      if (typeof nextProps.saveModel != "undefined") {
        this.setState({
          saveModel: {
            startDate: nextProps.saveModel.startDate,
            startTime: nextProps.saveModel.startTime,
            endDate: nextProps.saveModel.endDate,
            endTime: nextProps.saveModel.endTime
          }
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$state2 = this.state,
          startDateTip = _this$state2.startDateTip,
          endDateTip = _this$state2.endDateTip,
          startTimeTip = _this$state2.startTimeTip,
          endTimeTip = _this$state2.endTimeTip,
          startDateLabel = _this$state2.startDateLabel,
          startTimeLabel = _this$state2.startTimeLabel,
          endDateLabel = _this$state2.endDateLabel,
          endTimeLabel = _this$state2.endTimeLabel,
          saveModel = _this$state2.saveModel;
      console.log(saveModel, "saveModel in render");
      return /*#__PURE__*/React.createElement("div", {
        className: "promotionTimeRangeWrap"
      }, /*#__PURE__*/React.createElement(Row, {
        gutter: 30,
        style: {
          marginBottom: "10px"
        }
      }, /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, startDateLabel), /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, startTimeLabel), /*#__PURE__*/React.createElement(Col, {
        span: 1
      }, "\xA0"), /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, endDateLabel), /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, endTimeLabel)), /*#__PURE__*/React.createElement(Row, {
        gutter: 30,
        style: {
          marginBottom: "10px"
        }
      }, /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, /*#__PURE__*/React.createElement(ErrorTip, {
        message: startDateTip
      }, /*#__PURE__*/React.createElement(DatePicker, {
        onChange: function onChange(a, b) {
          return _this2.startDateChange(a, b);
        },
        suffixIcon: null,
        format: DateTimeFormat.dateFormat,
        value: saveModel.startDate ? moment(saveModel.startDate, DateTimeFormat.dateFormat) : ""
      }))), /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, /*#__PURE__*/React.createElement(ErrorTip, {
        message: startTimeTip
      }, /*#__PURE__*/React.createElement(TimePicker, {
        suffixIcon: null,
        format: DateTimeFormat.timeFormat,
        onChange: function onChange(a, b) {
          return _this2.startTimeChange(a, b);
        },
        className: "promotionAuditTimePicker",
        value: saveModel.startTime ? moment(saveModel.startTime, DateTimeFormat.timeFormat) : ""
      }))), /*#__PURE__*/React.createElement(Col, {
        span: 1
      }, " "), /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, /*#__PURE__*/React.createElement(ErrorTip, {
        message: endDateTip
      }, /*#__PURE__*/React.createElement(DatePicker, {
        suffixIcon: null,
        format: DateTimeFormat.dateFormat,
        onChange: function onChange(a, b) {
          return _this2.endDateChange(a, b);
        },
        value: saveModel.endDate ? moment(saveModel.endDate, DateTimeFormat.dateFormat) : ""
      }))), /*#__PURE__*/React.createElement(Col, {
        span: 4
      }, /*#__PURE__*/React.createElement(ErrorTip, {
        message: endTimeTip
      }, /*#__PURE__*/React.createElement(TimePicker, {
        suffixIcon: null,
        format: DateTimeFormat.timeFormat,
        onChange: function onChange(a, b) {
          return _this2.endTimeChange(a, b);
        },
        className: "promotionAuditTimePicker",
        value: saveModel.endTime ? moment(saveModel.endTime, DateTimeFormat.timeFormat) : ""
      })))));
    }
  }]);

  return TimeRangeSection;
}(React.Component);

export { TimeRangeSection as default };
