// @aurum/pfe-ui/es/cascader/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import { Cascader as AntdCascader, ConfigProvider } from 'antd';
import IconFont from '../common/IconFont';
import { IconCloseCircleFill } from '@aurum/icons';

var Index = function Index(props) {
  var _React$useContext = React.useContext(ConfigProvider.ConfigContext),
      getPrefixCls = _React$useContext.getPrefixCls;

  return /*#__PURE__*/React.createElement(AntdCascader, _extends({
    suffixIcon: /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-xiangxia",
      style: {
        fontSize: '12px'
      }
    }),
    expandIcon: /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-xiangxia",
      style: {
        fontSize: '12px'
      }
    }),
    clearIcon: /*#__PURE__*/React.createElement(IconCloseCircleFill, {
      className: "".concat(getPrefixCls(), "-clear-icon")
    })
  }, props, {
    className: "".concat(getPrefixCls(), "-cascader ").concat(getPrefixCls(), "-transform ").concat(props.className || '')
  }));
};

export default Index;
