// @mcd/portal-components/dist/components/download-modal/index.js
import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

import React, { PureComponent } from 'react';
import { Typography, Input, Space } from 'antd';
import Modal from '../../antd/modal';
import PropTypes from 'prop-types';
var TextArea = Input.TextArea;
var Text = Typography.Text,
    Link = Typography.Link;

var DownloadModal = /*#__PURE__*/function (_PureComponent) {
  _inherits(DownloadModal, _PureComponent);

  var _super = _createSuper(DownloadModal);

  function DownloadModal() {
    var _this;

    _classCallCheck(this, DownloadModal);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "overwriteProps", function () {
      var props = _objectSpread({}, _this.props);

      delete props.href;
      delete props.linkName;
      delete props.value;
      return props;
    });

    return _this;
  }

  _createClass(DownloadModal, [{
    key: "render",
    value: function render() {
      var props = this.overwriteProps();
      return /*#__PURE__*/React.createElement(Modal, _extends({
        className: "download-modal"
      }, props, {
        okText: window.$t ? $t('btn_confirm') : '确定',
        cancelText: window.$t ? $t('btn_cancel') : '取消'
      }), /*#__PURE__*/React.createElement(Space, {
        direction: "vertical"
      }, /*#__PURE__*/React.createElement(Text, null, "\u6587\u672C\u5185\u5BB9\u4F8B\uFF1A\u70B9\u51FB\u786E\u5B9A\u6309\u94AE\u540E\uFF0C\u8BF7\u81F3", this.props.href ? /*#__PURE__*/React.createElement(Link, {
        href: this.props.href
      }, this.props.linkName) : this.props.linkName, "\u4EFB\u52A1\u4E2D\u67E5\u770B\u5E76\u83B7\u53D6\u6587\u4EF6"), /*#__PURE__*/React.createElement(Text, null, "\u4E0B\u8F7D\u6587\u4EF6\u540D\uFF1A")), /*#__PURE__*/React.createElement(TextArea, {
        value: this.props.value
      }));
    }
  }]);

  return DownloadModal;
}(PureComponent);

DownloadModal.propTypes = {
  /** 跳转地址，比如：'/store/task/export' */
  href: PropTypes.string,

  /** 地址名称 */
  linkName: PropTypes.string,

  /** 文本框显示的内容 */
  value: PropTypes.string
};
DownloadModal.defaultProps = {};
DownloadModal.displayName = 'DownloadModal丨下载弹窗';
export default DownloadModal;
