// @aurum/pfe-ui/es/upload/SingleFileUpload.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { ConfigProvider, Upload } from 'antd';
import Button from '../button';
import IconFont from '../common/IconFont';
export default (function (props) {
  var _React$useContext = React.useContext(ConfigProvider.ConfigContext),
    getPrefixCls = _React$useContext.getPrefixCls;
  return /*#__PURE__*/React.createElement("div", {
    className: "".concat(getPrefixCls(), "-upload-single-file")
  }, /*#__PURE__*/React.createElement(Upload, _extends({
    progress: {
      strokeColor: '#ffbc0d',
      strokeWidth: 6
    },
    showUploadList: {
      showRemoveIcon: true,
      removeIcon: /*#__PURE__*/React.createElement(IconFont, {
        type: "icon-qingchu",
        className: "".concat(getPrefixCls(), "-upload-single-file-delete-icon")
      })
    },
    maxCount: 1
  }, props), /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-shangchuan"
    })
  }, "\u4E0A\u4F20\u6587\u4EF6")));
});
