// @aurum/pfe-ui/es/date-picker/DatePicker.js
import _extends from "@babel/runtime/helpers/esm/extends";
var __rest = this && this.__rest || function (s, e) {
  var t = {};
  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }
  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};
import React from 'react';
import IconFont from '../common/IconFont';
import { DatePicker as AntdDatePicker } from 'antd';
var DatePicker = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _props$superPrevIcon = props.superPrevIcon,
    superPrevIcon = _props$superPrevIcon === void 0 ? /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-wangqianfanye"
    }) : _props$superPrevIcon,
    _props$superNextIcon = props.superNextIcon,
    superNextIcon = _props$superNextIcon === void 0 ? /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-wanghoufanye"
    }) : _props$superNextIcon,
    _props$prevIcon = props.prevIcon,
    prevIcon = _props$prevIcon === void 0 ? /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-fanhui"
    }) : _props$prevIcon,
    _props$nextIcon = props.nextIcon,
    nextIcon = _props$nextIcon === void 0 ? /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-tiaozhuan"
    }) : _props$nextIcon,
    _props$suffixIcon = props.suffixIcon,
    suffixIcon = _props$suffixIcon === void 0 ? /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-shijian"
    }) : _props$suffixIcon,
    rest = __rest(props, ["superPrevIcon", "superNextIcon", "prevIcon", "nextIcon", "suffixIcon"]);
  return /*#__PURE__*/React.createElement(AntdDatePicker, _extends({}, rest, {
    suffixIcon: suffixIcon,
    superPrevIcon: superPrevIcon,
    superNextIcon: superNextIcon,
    prevIcon: prevIcon,
    nextIcon: nextIcon,
    ref: ref
  }));
});
DatePicker.WeekPicker = AntdDatePicker.WeekPicker;
DatePicker.MonthPicker = AntdDatePicker.MonthPicker;
DatePicker.YearPicker = AntdDatePicker.YearPicker;
DatePicker.RangePicker = AntdDatePicker.RangePicker;
DatePicker.TimePicker = AntdDatePicker.TimePicker;
DatePicker.QuarterPicker = AntdDatePicker.QuarterPicker;
export default DatePicker;
