// @aurum/pfe-ui/es/popconfirm/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { ConfigProvider, Popconfirm as AntdPopconfirm } from 'antd';
var Popconfirm = function Popconfirm(props) {
  var _React$useContext = React.useContext(ConfigProvider.ConfigContext),
    getPrefixCls = _React$useContext.getPrefixCls;
  props = _extends(_extends({}, props), {
    okType: 'default',
    cancelButtonProps: {
      type: 'primary'
    },
    mcdType: (props === null || props === void 0 ? void 0 : props.mcdType) || '',
    icon: /* props?.icon || */'',
    onCancel: props === null || props === void 0 ? void 0 : props.onConfirm,
    onConfirm: props === null || props === void 0 ? void 0 : props.onCancel,
    cancelText: (props === null || props === void 0 ? void 0 : props.okText) || '确认',
    okText: (props === null || props === void 0 ? void 0 : props.cancelText) || '取消'
  });
  var isTableType = (props === null || props === void 0 ? void 0 : props.mcdType) === 'table';
  return /*#__PURE__*/React.createElement(ConfigProvider, {
    autoInsertSpaceInButton: false
  }, /*#__PURE__*/React.createElement(AntdPopconfirm, _extends({
    overlayClassName: isTableType ? "".concat(getPrefixCls(), "-modal-popconfirm") : ''
  }, props)));
};
export default Popconfirm;
