// @aurum/pfe-ui/es/side-bar/SideBar.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import helper from '../common/helper';
import { IconSearch } from '@aurum/icons';
import { Input, Layout } from '../index';
var Search = Input.Search;

var SideBar = /*#__PURE__*/function (_React$Component) {
  _inherits(SideBar, _React$Component);

  var _super = _createSuper(SideBar);

  function SideBar() {
    _classCallCheck(this, SideBar);

    return _super.apply(this, arguments);
  }

  _createClass(SideBar, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          _this$props$className = _this$props.className,
          className = _this$props$className === void 0 ? '' : _this$props$className,
          _this$props$collapsed = _this$props.collapsed,
          collapsed = _this$props$collapsed === void 0 ? false : _this$props$collapsed,
          _this$props$logoTitle = _this$props.logoTitle,
          logoTitle = _this$props$logoTitle === void 0 ? '' : _this$props$logoTitle,
          _this$props$logoTitle2 = _this$props.logoTitleImage,
          logoTitleImage = _this$props$logoTitle2 === void 0 ? '' : _this$props$logoTitle2,
          _this$props$searchPla = _this$props.searchPlaceHolder,
          searchPlaceHolder = _this$props$searchPla === void 0 ? '' : _this$props$searchPla,
          _this$props$onMenuSea = _this$props.onMenuSearch,
          onMenuSearch = _this$props$onMenuSea === void 0 ? function () {} : _this$props$onMenuSea,
          _this$props$treeConte = _this$props.treeContent,
          treeContent = _this$props$treeConte === void 0 ? function () {
        return null;
      } : _this$props$treeConte,
          _this$props$menuConte = _this$props.menuContent,
          menuContent = _this$props$menuConte === void 0 ? function () {
        return null;
      } : _this$props$menuConte;
      return /*#__PURE__*/React.createElement(Layout.Sider, {
        className: className,
        collapsed: collapsed,
        collapsedWidth: collapsed ? 64 : 208,
        width: collapsed ? 64 : 208
      }, /*#__PURE__*/React.createElement("div", {
        className: "logo-wrapper"
      }, /*#__PURE__*/React.createElement("img", {
        src: helper.getImageUrl('/logo/logo-image.png'),
        className: "logo-image",
        alt: ""
      }), !collapsed && (logoTitle ? /*#__PURE__*/React.createElement("div", {
        className: "logo-title-text"
      }, logoTitle) : /*#__PURE__*/React.createElement("img", {
        src: logoTitleImage || helper.getImageUrl('/logo/logo-title.png'),
        className: "logo-title-image",
        alt: ""
      }))), !collapsed && /*#__PURE__*/React.createElement("div", {
        className: "menu-search"
      }, /*#__PURE__*/React.createElement(Search, {
        bordered: false,
        placeholder: searchPlaceHolder,
        onSearch: onMenuSearch,
        onChange: function onChange(e) {
          var _a;

          return onMenuSearch((_a = e === null || e === void 0 ? void 0 : e.target) === null || _a === void 0 ? void 0 : _a.value);
        },
        suffix: /*#__PURE__*/React.createElement(IconSearch, null)
      }), treeContent()), menuContent());
    }
  }]);

  return SideBar;
}(React.Component);

export default SideBar;
