// @mcd/portal-components/dist/components/dict-checkbox/index.js
import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'antd';
import { getLocalItem, Constants } from '../../utils/cache';
import { getDictionary } from '../../utils/DictUtil';
import { getDictList } from '../../utils/common';

var DictCheckbox = /*#__PURE__*/function (_Component) {
  _inherits(DictCheckbox, _Component);

  var _super = _createSuper(DictCheckbox);

  function DictCheckbox(props) {
    var _this;

    _classCallCheck(this, DictCheckbox);

    _this = _super.call(this, props);

    _defineProperty(_assertThisInitialized(_this), "getList", function (data) {
      _this.setState({
        typeList: getDictList(_this.props.type, data)
      });
    });

    _this.state = {
      typeList: []
    };
    return _this;
  }

  _createClass(DictCheckbox, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this2 = this;

      if (!this.props.options) {
        var data = getLocalItem(Constants.DICTIONARYLIST) || {};

        if (data && Object.keys(data).length > 0) {
          this.getList(data);
        } else {
          getDictionary('DICTIONARYLIST').then(function (res) {
            _this2.getList(res);
          });
        }
      }
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(Checkbox.Group, _extends({}, this.props, {
        options: this.props.options || this.state.typeList
      }));
    }
  }]);

  return DictCheckbox;
}(Component);

DictCheckbox.propTypes = {
  /** 字典类型 */
  type: PropTypes.string.isRequired,

  /** 指定选中的选项 */
  value: PropTypes.array,

  /** 默认值 */
  defaultValue: PropTypes.array,

  /** 选择后的返回值 */
  onChange: PropTypes.func,

  /** 自定义选项 */
  options: PropTypes.array
};
DictCheckbox.displayName = 'DictCheckbox丨字典多选框';
export default DictCheckbox;
