// @mcd/portal-components/dist/utils/FileUtil.js
import { getValueByKey } from './common';
import { getToken } from './auth';
import { get } from './http';
import { ApiConstants } from './ApiConstants'; // 获取文件预览的地址

export function getPreviewUrl(docId) {
  var docViewUrl = getValueByKey('doc.preview.url');
  return docViewUrl + '/idocview/onlinePreview' + '?id=' + docId + '&token=' + getToken();
} // 下载文件

export function downLoadFileById(id) {
  return window.open(ApiConstants.DOWNLOAD_DOC_URL + '?id=' + id + '&token=' + getToken());
} // 获取文件详细信息

export function getFileDetailById(id) {
  var params = {
    id: id
  };
  var promise = get(ApiConstants.DOC_DETAIL + '?token=' + getToken(), params, '/api/inner/docmgmt');
  return promise;
} // 获取上传文件的信息
// fileProp: 获取下载地址是否永久有效 0:有效时间30分钟,1:永久有效

export function getUploadInfo() {
  var appCode = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'portal';
  var fileProp = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : '1';
  var dirList = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : '';
  var isRandom = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : '1';
  return {
    name: 'file',
    action: ApiConstants.UPLOAD_DOC_URL,
    data: {
      appCode: appCode,
      fileProp: fileProp,
      dirList: dirList,
      isRandom: isRandom
    },
    headers: {
      Authorization: getToken()
    }
  };
}
