// @mcd/portal-components/dist/utils/format.js
export var dateFormat_ = "YYYY-MM-DD";
export var dateFormat = "YYYY/MM/DD";
export var timeFormat = "HH:mm:ss";
export var timeFormat_ = "HH:mm";
export var dateAndTime = "YYYY/MM/DD HH:mm:ss";
export var dateAndTime_ = "YYYY-MM-DD HH:mm:ss";
/**
 *
 * @param {*} time
 */

export var dateAndTime_Fn = function dateAndTime_Fn(time) {
  return time + " 00:00:00";
};
/**
 *
 * @param {*} dateTimeMoment
 * @param {*} formatStr
 */

export var dateTimeToString = function dateTimeToString(dateTimeMoment, formatStr) {
  var type = Object.prototype.toString.call(dateTimeMoment);

  if (type.toLowerCase().indexOf("string") > -1) {
    return dateTimeMoment;
  }

  var year = dateTimeMoment.get("year");
  var month = dateTimeMoment.get("month") + 1;
  var day = dateTimeMoment.get("date");
  var hour = dateTimeMoment.get("hours");
  var minute = dateTimeMoment.get("minutes");
  var second = dateTimeMoment.get("seconds");

  switch (formatStr) {
    case "YYYY-MM-DD":
      return "".concat(year, "-").concat(month < 10 ? "0" + month : month, "-").concat(day < 10 ? "0" + day : day);

    case "HH:mm:ss":
      return "".concat(hour < 10 ? "0" + hour : hour, ":").concat(minute < 10 ? "0" + minute : minute, ":").concat(second < 10 ? "0" + second : second);
  }
};
/**
 *
 * @param {*} str
 */

export var getTimestamp = function getTimestamp(str) {
  return str ? new Date(str).getTime() : "";
};
