// @omc/boss-https/lib/index.js
import request from './request';

const BASE_API_URL = '/';
export default function (config = {}) {
  const requestCtx = request(config);
  function send(url, options) {
    options = options || {};
    let { params, query, baseUrl, pathVars, headers, loading, method, responseType } = { ...options };
    if (loading === undefined) loading = true;

    const reqArgs = {
      url: (baseUrl || config.baseUrl || BASE_API_URL) + url,
      method: method || 'get',
      pathVars,
      params: {},
      headers
    };

    if (responseType) reqArgs.responseType = responseType;
    if (method === 'get') {
      reqArgs.params = params;
    } else {
      if (query) {
        reqArgs.params = query;
      }
      reqArgs.data = params;
    }

    return new Promise((resolve, reject) => {
      if (config.openLoading && loading && window.__store) {
        window.__store.dispatch(config.openLoading());
      }
      requestCtx(reqArgs).then((response) => {
        resolve(response); // 返回成功
      }).catch((e) => {
        reject(e);
      });
    });
  }

  return {
    send,
    get: (url, options) => {
      options.method = 'get';
      return send(url, options);
    },
    post: (url, options) => {
      options.method = 'post';
      return send(url, options);
    },
    put: (url, options) => {
      options.method = 'put';
      return send(url, options);
    },
    delete: (url, options) => {
      options.method = 'delete';
      return send(url, options);
    }
  }
}
