// @mcd/portal-components/dist/dynamic/dynamicItem/dynamicForm.js
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import _regeneratorRuntime from "@babel/runtime/regenerator";

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

/**
 * @param  {Props} modelCode formCode
 * @return {Function} validation function
 * @author ethan.yan
 */
import React, { Component } from 'react';
import { Row, Col, Tabs, Divider, Form, Button, message, Spin } from '../../index';
import { withRouter } from 'react-router-dom';
import moment from 'moment';
import { withTranslation } from 'react-i18next';
import { getFormGroupDetail, createForDE } from '../../api/mek/dataformAPI';
import { getDictionaryListByType } from '../../utils/DictUtil';
import * as formItem from './formItem';
var TabPane = Tabs.TabPane;

var DynamicForm = /*#__PURE__*/function (_Component) {
  _inherits(DynamicForm, _Component);

  var _super = _createSuper(DynamicForm);

  function DynamicForm(_props) {
    var _this;

    _classCallCheck(this, DynamicForm);

    _this = _super.call(this, _props);

    _defineProperty(_assertThisInitialized(_this), "formRef", /*#__PURE__*/React.createRef());

    _defineProperty(_assertThisInitialized(_this), "state", {
      // tab切换的key
      tabActiveKey: '',
      // tab的数据
      tabList: [],
      // 对应tab的相关表单
      tabFormList: {},
      // 是否显示tab
      showTab: false,
      // 表单分组
      groupIndexList: [],
      disabledEnum: {},
      // 时间控件的code
      dateCodeList: [],
      // 页面是否加载中
      loading: false,
      // 按钮置灰
      btnLoading: false
    });

    _defineProperty(_assertThisInitialized(_this), "initData", /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var res;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _this.setState({
                loading: true
              });

              _context.prev = 1;
              _context.next = 4;
              return getFormGroupDetail({
                modelCode: _this.props.modelCode,
                formCode: _this.props.formCode || ''
              });

            case 4:
              res = _context.sent;

              _this.adapterData(res.data);

              _context.next = 11;
              break;

            case 8:
              _context.prev = 8;
              _context.t0 = _context["catch"](1);
              message.error('MEK获取数据失败');

            case 11:
              _this.setState({
                loading: false
              });

            case 12:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, null, [[1, 8]]);
    })));

    _defineProperty(_assertThisInitialized(_this), "adapterData", function (data) {
      var _data$;

      var disabledEnum = {};

      if ((_data$ = data[0]) !== null && _data$ !== void 0 && _data$.propDisabledJson) {
        try {
          var _data$2;

          disabledEnum = JSON.parse((_data$2 = data[0]) === null || _data$2 === void 0 ? void 0 : _data$2.propDisabledJson);
        } catch (_unused) {
          message.error('JSON 解析错误');
        }
      }

      var tabList = data.map(function (item) {
        return {
          tab: item.groupName,
          key: item.groupId,
          language: item.groupNameEncoding
        };
      });
      var showTab = data.length === 1 ? data[0].showTab : true;
      var tabFormList = {};
      var groupIndexList = {};
      var dateCodeList = [];
      data.forEach(function (item) {
        var groupIndexArr = [];
        tabFormList[item.groupId] = item.groupModelList.map(function (item, index) {
          item.itemModelList = item.itemModelList.filter(function (item) {
            var _item$property;

            return !_this.props.hiddenFields.includes(item === null || item === void 0 ? void 0 : (_item$property = item.property) === null || _item$property === void 0 ? void 0 : _item$property.code);
          });

          if (index === 0) {
            groupIndexArr[index] = item.itemModelList.length;
          } else {
            groupIndexArr[index] = Number.parseInt(item.itemModelList.length) + Number.parseInt(groupIndexArr[index - 1]);
          }

          return item.itemModelList.map(function (itemMode) {
            var _itemMode$property;

            var property = _this._getFormAttr(itemMode, disabledEnum);

            if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property = itemMode.property) === null || _itemMode$property === void 0 ? void 0 : _itemMode$property.type) === '4') {
              var _itemMode$property2;

              dateCodeList.push(itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property2 = itemMode.property) === null || _itemMode$property2 === void 0 ? void 0 : _itemMode$property2.code);
            }

            return _objectSpread(_objectSpread({}, property), {}, {
              columnNum: item.columnNum,
              groupName: item.groupName,
              groupKey: item.groupId,
              groupLanguage: item.groupNameEncoding
            });
          });
        }).flat();
        groupIndexList[item.groupId] = groupIndexArr;
      });

      _this.setState({
        tabList: tabList,
        showTab: showTab,
        tabFormList: tabFormList,
        groupIndexList: groupIndexList,
        dateCodeList: dateCodeList
      });
    });

    _defineProperty(_assertThisInitialized(_this), "_getFormAttr", function (itemMode, disabledEnum) {
      var _disabledEnum$itemMod, _itemMode$property3, _itemMode$property4, _itemMode$property5, _itemMode$property6, _itemMode$property7, _itemMode$property8, _itemMode$property9, _itemMode$property10, _itemMode$property11, _itemMode$property12, _itemMode$property13, _itemMode$property14, _itemMode$property15, _itemMode$property16, _itemMode$property17, _itemMode$property18, _itemMode$property19, _itemMode$property20, _filter$, _itemMode$property21, _itemMode$property22, _filter$2, _itemMode$property23, _itemMode$property24, _itemMode$property25, _itemMode$property26, _itemMode$property27, _itemMode$property28, _itemMode$property29, _itemMode$property30, _itemMode$property31, _itemMode$property33, _itemMode$property34, _itemMode$property35, _itemMode$property36, _itemMode$property37, _itemMode$property38, _itemMode$property39, _itemMode$property40;

      var property = {}; // 0：单行文本，1：多行文本，2：数字，3：下拉框，4：日期，5：图片，6：附件，7：关联属性，8：公式计算'

      property['disabled'] = Boolean((_disabledEnum$itemMod = disabledEnum[itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property3 = itemMode.property) === null || _itemMode$property3 === void 0 ? void 0 : _itemMode$property3.code]) === null || _disabledEnum$itemMod === void 0 ? void 0 : _disabledEnum$itemMod.disabled);
      property['type'] = itemMode === null || itemMode === void 0 ? void 0 : itemMode.type; // property['label'] = itemMode?.property?.name;

      property['name'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property4 = itemMode.property) === null || _itemMode$property4 === void 0 ? void 0 : _itemMode$property4.code;
      property['formType'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property5 = itemMode.property) === null || _itemMode$property5 === void 0 ? void 0 : _itemMode$property5.type;
      property['validType'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property6 = itemMode.property) === null || _itemMode$property6 === void 0 ? void 0 : _itemMode$property6.validType;
      property['validPattern'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property7 = itemMode.property) === null || _itemMode$property7 === void 0 ? void 0 : _itemMode$property7.validPattern;
      property['validMessage'] = _this.props.t(itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property8 = itemMode.property) === null || _itemMode$property8 === void 0 ? void 0 : _itemMode$property8.validMessage);
      property['maxLength'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property9 = itemMode.property) === null || _itemMode$property9 === void 0 ? void 0 : _itemMode$property9.maxLength;
      property['maxValue'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property10 = itemMode.property) === null || _itemMode$property10 === void 0 ? void 0 : _itemMode$property10.maxValue;
      property['minValue'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property11 = itemMode.property) === null || _itemMode$property11 === void 0 ? void 0 : _itemMode$property11.minValue;
      property['precision'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property12 = itemMode.property) === null || _itemMode$property12 === void 0 ? void 0 : _itemMode$property12.decimalPrecision;
      property['required'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property13 = itemMode.property) === null || _itemMode$property13 === void 0 ? void 0 : _itemMode$property13.notNullFlag;
      property['extra'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property14 = itemMode.property) === null || _itemMode$property14 === void 0 ? void 0 : _itemMode$property14.tips;
      property['label'] = _this.props.t(itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property15 = itemMode.property) === null || _itemMode$property15 === void 0 ? void 0 : _itemMode$property15.languageEncoding);
      property['component'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property16 = itemMode.property) === null || _itemMode$property16 === void 0 ? void 0 : _itemMode$property16.component;

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property17 = itemMode.property) === null || _itemMode$property17 === void 0 ? void 0 : _itemMode$property17.validType) === '2') {
        property['validMessage'] = _this.props.t('portal_mobile_number_valid');
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property18 = itemMode.property) === null || _itemMode$property18 === void 0 ? void 0 : _itemMode$property18.validType) === '3') {
        property['validMessage'] = _this.props.t('portal_email_valid');
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property19 = itemMode.property) === null || _itemMode$property19 === void 0 ? void 0 : _itemMode$property19.validType) === '4') {
        property['validMessage'] = _this.props.t('portal_id_card_valid');
      }

      property['addonBefore'] = ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property20 = itemMode.property) === null || _itemMode$property20 === void 0 ? void 0 : _itemMode$property20.prefixConfig) || []).map(function (item) {
        return {
          label: item.name,
          value: item.name
        };
      });
      property['addonBeforeDefault'] = (_filter$ = ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property21 = itemMode.property) === null || _itemMode$property21 === void 0 ? void 0 : _itemMode$property21.prefixConfig) || []).filter(function (item) {
        return item.ifDefault;
      })[0]) === null || _filter$ === void 0 ? void 0 : _filter$.name;
      property['addonAfter'] = ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property22 = itemMode.property) === null || _itemMode$property22 === void 0 ? void 0 : _itemMode$property22.suffixConfig) || []).map(function (item) {
        return {
          label: item.name,
          value: item.name
        };
      });
      property['addonAfterDefault'] = (_filter$2 = ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property23 = itemMode.property) === null || _itemMode$property23 === void 0 ? void 0 : _itemMode$property23.suffixConfig) || []).filter(function (item) {
        return item.ifDefault;
      })[0]) === null || _filter$2 === void 0 ? void 0 : _filter$2.name;
      (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property24 = itemMode.property) === null || _itemMode$property24 === void 0 ? void 0 : _itemMode$property24.selectorType) === '3' ? property['isMultipleSelect'] = true : property['isMultipleSelect'] = false;
      property['dateFormat'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property25 = itemMode.property) === null || _itemMode$property25 === void 0 ? void 0 : _itemMode$property25.dateFormat;
      property['maxFileSize'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property26 = itemMode.property) === null || _itemMode$property26 === void 0 ? void 0 : _itemMode$property26.maxFileSize;
      property['maxFileNum'] = itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property27 = itemMode.property) === null || _itemMode$property27 === void 0 ? void 0 : _itemMode$property27.amount;
      property['defaultValue'] = (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property28 = itemMode.property) === null || _itemMode$property28 === void 0 ? void 0 : _itemMode$property28.defaultValue) || _this.props.initialValues[itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property29 = itemMode.property) === null || _itemMode$property29 === void 0 ? void 0 : _itemMode$property29.code];
      property['optionList'] = ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property30 = itemMode.property) === null || _itemMode$property30 === void 0 ? void 0 : _itemMode$property30.propertyOptionList) || []).map(function (item) {
        return {
          label: item.label || item.value,
          value: item.value
        };
      });

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property31 = itemMode.property) === null || _itemMode$property31 === void 0 ? void 0 : _itemMode$property31.optionDataSource) === '2') {
        var _itemMode$property32;

        property['optionList'] = (getDictionaryListByType(itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property32 = itemMode.property) === null || _itemMode$property32 === void 0 ? void 0 : _itemMode$property32.dictName) || []).map(function (item) {
          return {
            label: item.dictLabelCn,
            value: item.dictValue
          };
        });
      }

      if (itemMode.type === '2') {
        property['formType'] = 'slot';
        property['name'] = itemMode.code;
        property['soltRow'] = itemMode.rows;
        property['soltColumn'] = itemMode.column;
      } // selectorType  1-单选  2-复选框  否则是下拉（0-下拉单选 3-下拉多选）


      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property33 = itemMode.property) === null || _itemMode$property33 === void 0 ? void 0 : _itemMode$property33.type) === '3' && (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property34 = itemMode.property) === null || _itemMode$property34 === void 0 ? void 0 : _itemMode$property34.selectorType) === '1') {
        property['formType'] = '3_1';
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property35 = itemMode.property) === null || _itemMode$property35 === void 0 ? void 0 : _itemMode$property35.type) === '3' && (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property36 = itemMode.property) === null || _itemMode$property36 === void 0 ? void 0 : _itemMode$property36.selectorType) === '2') {
        property['formType'] = '3_2';
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property37 = itemMode.property) === null || _itemMode$property37 === void 0 ? void 0 : _itemMode$property37.type) === '4' && (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property38 = itemMode.property) === null || _itemMode$property38 === void 0 ? void 0 : _itemMode$property38.selectorType) === '1') {
        property['formType'] = '4_1';
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property39 = itemMode.property) === null || _itemMode$property39 === void 0 ? void 0 : _itemMode$property39.type) === '4' && (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property40 = itemMode.property) === null || _itemMode$property40 === void 0 ? void 0 : _itemMode$property40.selectorType) === '2') {
        property['isShowTime'] = true;
      }

      return property;
    });

    _defineProperty(_assertThisInitialized(_this), "_divider", function (props) {
      var t = _this.props.t;
      return /*#__PURE__*/React.createElement(Col, {
        span: 24,
        key: "".concat(props.groupKey, "group")
      }, /*#__PURE__*/React.createElement(Divider, null), /*#__PURE__*/React.createElement("h2", {
        style: {
          fontSize: '14px',
          color: '#333333',
          fontWeight: 'bold',
          marginBottom: '20px'
        }
      }, t(props === null || props === void 0 ? void 0 : props.groupLanguage) || ''));
    });

    _defineProperty(_assertThisInitialized(_this), "_renderForm", function (props, tabKey) {
      var groupIndexList = _this.state.groupIndexList;

      if (!props) {
        return;
      }

      return /*#__PURE__*/React.createElement(Row, {
        gutter: 30
      }, props.map(function (item, index) {
        return [groupIndexList[tabKey].includes(index) ? _this._divider(item) : '', formItem["formItem".concat(item.formType)](_objectSpread({}, item), _this.props.children)];
      }));
    });

    _defineProperty(_assertThisInitialized(_this), "_renderTabs", function (props) {
      var t = _this.props.t;
      return /*#__PURE__*/React.createElement(Tabs, null, props.map(function (item) {
        return /*#__PURE__*/React.createElement(TabPane, {
          tab: t(item.language),
          key: item.key
        }, _this._renderForm(_this.state.tabFormList[item.key], item.key));
      }));
    });

    _defineProperty(_assertThisInitialized(_this), "_changeValues", function (values) {
      _this.state.dateCodeList.forEach(function (code) {
        if (values[code]) {
          values[code] = moment(values[code]).format('YYYY-MM-DD HH:mm:ss');
        }
      });

      return _objectSpread({}, values);
    });

    _defineProperty(_assertThisInitialized(_this), "_handleSubmit", /*#__PURE__*/function () {
      var _ref2 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee2(values) {
        var params;
        return _regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                params = _this._changeValues(values);

                if (!(typeof _this.props.onFinish === 'function')) {
                  _context2.next = 9;
                  break;
                }

                _this.setState({
                  btnLoading: true
                });

                _context2.next = 5;
                return _this.props.onFinish(params);

              case 5:
                _this.setState({
                  btnLoading: false
                });

                return _context2.abrupt("return");

              case 9:
                _this._createForDE(params);

              case 10:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2);
      }));

      return function (_x) {
        return _ref2.apply(this, arguments);
      };
    }());

    _defineProperty(_assertThisInitialized(_this), "_createForDE", /*#__PURE__*/function () {
      var _ref3 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee3(values) {
        var params, res;
        return _regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                params = {
                  'props': values,
                  formCode: _this.props.formCode,
                  modelCode: _this.props.modelCode
                };
                _context3.prev = 1;
                _context3.next = 4;
                return createForDE(params);

              case 4:
                res = _context3.sent;
                message.success('新增成功');
                _context3.next = 11;
                break;

              case 8:
                _context3.prev = 8;
                _context3.t0 = _context3["catch"](1);
                message.error('新增失败');

              case 11:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, null, [[1, 8]]);
      }));

      return function (_x2) {
        return _ref3.apply(this, arguments);
      };
    }());

    _defineProperty(_assertThisInitialized(_this), "_handleGoBack", function () {
      if (typeof _this.props.onGoBack === 'function') {
        _this.props.onGoBack();

        return;
      }

      _this.props.history.goBack();
    });

    _defineProperty(_assertThisInitialized(_this), "_handleFinishFailed", function (_ref4) {
      var values = _ref4.values,
          errorFields = _ref4.errorFields,
          outOfDate = _ref4.outOfDate;

      _this.formRef.current.scrollToField(errorFields[0]['name']);
    });

    return _this;
  }

  _createClass(DynamicForm, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.initData();
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (this.props.modelCode !== prevProps.modelCode || this.props.formCode !== prevProps.formCode) {
        this.initData();
      }
    } // 向后端请求数据

  }, {
    key: "render",
    value: function render() {
      var _tabList$, _tabList$2;

      var props = this.props;
      var _this$state = this.state,
          showTab = _this$state.showTab,
          tabList = _this$state.tabList,
          tabFormList = _this$state.tabFormList;
      return /*#__PURE__*/React.createElement("div", {
        style: {
          background: '#fff',
          padding: '20px'
        }
      }, /*#__PURE__*/React.createElement(Spin, {
        spinning: this.state.loading,
        delay: 500
      }, /*#__PURE__*/React.createElement(Form, {
        onFinish: this._handleSubmit,
        onFinishFailed: this._handleFinishFailed,
        ref: this.formRef,
        layout: "vertical",
        colon: true
      }, showTab ? this._renderTabs(tabList) : this._renderForm(tabFormList[(_tabList$ = tabList[0]) === null || _tabList$ === void 0 ? void 0 : _tabList$.key], (_tabList$2 = tabList[0]) === null || _tabList$2 === void 0 ? void 0 : _tabList$2.key), (tabList.length > 0 || tabFormList.length > 0) && /*#__PURE__*/React.createElement(Form.Item, null, /*#__PURE__*/React.createElement(Button, {
        type: "primary",
        htmlType: "submit",
        loading: this.state.btnLoading,
        disabled: this.state.btnLoading
      }, "\u4FDD\u5B58"), /*#__PURE__*/React.createElement(Button, {
        type: "default",
        onClick: this._handleGoBack
      }, "\u8FD4\u56DE")))));
    }
  }]);

  return DynamicForm;
}(Component);

DynamicForm.defaultProps = {
  modelCode: '',
  formCode: '',
  initialValues: {},
  hiddenFields: []
};
export default withTranslation()(withRouter(DynamicForm));
