// @mcd/portal-components/dist/dynamic/dynamicItem/dynamicTable.js
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import _regeneratorRuntime from "@babel/runtime/regenerator";

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

/**
 * @param  {Props} modelCode formCode
 * @return {Function} validation function
 * @author ethan.yan
 */
import React, { PureComponent } from 'react';
import { Row, Col, Form, Button, message, Table, Divider, Popconfirm, Checkbox } from '../../index';
import { Link, withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import moment from 'moment';
import { UpOutlined, DownOutlined, QuestionCircleOutlined, ReloadOutlined } from '@ant-design/icons';
import { getTableDynamic } from '../../api/mek/dataformAPI';
import { getDictionaryListByType, getDictionaryLabel } from '../../utils/DictUtil';
import * as fetch from '../../utils/http';
import * as formItem from './formItem';
import './index.css';
var sortEnum = {
  '0': 'asc',
  '1': 'desc',
  'ascend': 'asc',
  'descend': 'desc'
};
var rowSelectionTypeEnum = {
  1: 'checkbox',
  2: 'radio'
};

function getTableHeight(pageClassName) {
  var otherHeight = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 171;

  if (!pageClassName) {
    return 0;
  } // 搜索区的高度


  var antForm = document.querySelector(".".concat(pageClassName, " .ant-form")); // 列表上面筛选组件的高度

  var tableTop = document.querySelector(".".concat(pageClassName, " .mek-table-top"));
  var autoHeight = otherHeight;
  antForm && (autoHeight += antForm.clientHeight);
  tableTop && (autoHeight += tableTop.clientHeight);
  return autoHeight;
}

var DynamicTable = /*#__PURE__*/function (_PureComponent) {
  _inherits(DynamicTable, _PureComponent);

  var _super = _createSuper(DynamicTable);

  function DynamicTable() {
    var _this;

    _classCallCheck(this, DynamicTable);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "formRef", /*#__PURE__*/React.createRef());

    _defineProperty(_assertThisInitialized(_this), "state", {
      tableClassName: _this.props.modelCode + _this.props.formCode,
      tableHeight: '',
      allColumnList: [],
      columnList: [],
      filterColumnOptionList: [],
      filterColumnValueList: [],
      formList: [],
      queryBtnUrl: '',
      createLink: '',
      deleteUrl: '',
      sourceList: [],
      isOpenCreate: false,
      isOpenEdit: false,
      isOpenQuery: false,
      isOpenDelete: false,
      isShowOption: true,
      // 页面是否加载中
      isLoading: true,
      isPagination: true,
      current: 1,
      total: 0,
      pageSize: 50,
      isShow: false,
      defaultSortFields: [],
      fieldOrder: [],
      rowSelectionType: '',
      selectedRowKeys: [],
      queryFetchMethod: 'get',
      deleteFetchMethod: 'post',
      formItemCode: [],
      sortSplitStr: ' '
    });

    _defineProperty(_assertThisInitialized(_this), "componentDidMount", function () {
      _this.initData();
    });

    _defineProperty(_assertThisInitialized(_this), "initData", /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var t, resTable;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              t = _this.props.t;
              _context.prev = 1;
              _context.next = 4;
              return getTableDynamic({
                modelCode: _this.props.modelCode,
                formCode: _this.props.formCode || ''
              });

            case 4:
              resTable = _context.sent;

              _this.adapterData(resTable.data);

              _context.next = 11;
              break;

            case 8:
              _context.prev = 8;
              _context.t0 = _context["catch"](1);
              message.error(t('portal_fetch_data_err'));

            case 11:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, null, [[1, 8]]);
    })));

    _defineProperty(_assertThisInitialized(_this), "columnRenderEnum", {
      4: function _(text) {
        return moment(text).format('YYYY-MM-DD HH:mm:ss');
      },
      32: function _(text, record, dictName) {
        return dictName ? getDictionaryLabel(dictName, text) : '';
      }
    });

    _defineProperty(_assertThisInitialized(_this), "adapterData", function () {
      var _arg$customFromConfig, _arg$customFromConfig2, _arg$customFromConfig3, _arg$customFromConfig4, _arg$customFromConfig5, _arg$customFromConfig6, _arg$customFromConfig7, _arg$customFromConfig8, _arg$customFromConfig9, _arg$customFromConfig10, _arg$customFromConfig11, _arg$customFromConfig12, _arg$customFromConfig13, _arg$customFromConfig14, _arg$customFromConfig15, _arg$customFromConfig16, _arg$customFromConfig17, _arg$customFromConfig18, _arg$customFromConfig19, _arg$customFromConfig20, _arg$customFromConfig21, _arg$customFromConfig22, _this$props, _this$props$initialVa, _this$props3, _this$props3$initialV, _this$props5, _this$props5$initialV;

      var arg = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      var t = _this.props.t;
      var lockColumn = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig = arg.customFromConfig) === null || _arg$customFromConfig === void 0 ? void 0 : _arg$customFromConfig.lockColumn;
      var columnList = arg['displayColumns'].map(function (item, index) {
        var random = Math.random();
        var columnTemp = {
          title: t(item.displayLanguageEncoding || item.languageEncoding),
          dataIndex: item.code || random,
          key: item.code || random,
          width: Number(item.widthFactor) || 'auto',
          sorter: Boolean(Number(item === null || item === void 0 ? void 0 : item.sortableFlag)),
          ellipsis: true
        };

        if (item.displayColumnType === '0' && item.type === '4' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '0') {
          columnTemp['render'] = _this.columnRenderEnum["".concat(item.type).concat(item.optionDataSource || '')];
        }

        if (item.displayColumnType === '0' && item.optionDataSource === '2' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '0') {
          columnTemp['render'] = function (text, record) {
            return _this.columnRenderEnum["".concat(item.type).concat(item.optionDataSource || '')](text, record, item.dictName);
          };
        }

        if (item.displayColumnType === '0' && item.type === '4' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '1') {
          columnTemp['render'] = _this.columnRenderEnum["".concat(item.type).concat(item.optionDataSource || '')];
        }

        if (item.displayColumnType === '0' && item.optionDataSource === '2' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '1') {
          columnTemp['render'] = function (text, record) {
            return /*#__PURE__*/React.createElement(Link, {
              to: "".concat(item === null || item === void 0 ? void 0 : item.routingAddress, "?").concat(((item === null || item === void 0 ? void 0 : item.hyperlinkParam) || []).map(function (key) {
                return "".concat(key, "=").concat(record[key]);
              }).join('&'))
            }, _this.columnRenderEnum["".concat(item.type).concat(item.optionDataSource)](text, record, item.dictName));
          };
        }

        if (item.displayColumnType === '0' && item.type !== '4' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '1' && item.optionDataSource !== '2') {
          columnTemp['render'] = function (text, record) {
            return /*#__PURE__*/React.createElement(Link, {
              to: "".concat(item === null || item === void 0 ? void 0 : item.routingAddress, "?").concat(((item === null || item === void 0 ? void 0 : item.hyperlinkParam) || []).map(function (key) {
                return "".concat(key, "=").concat(record[key]);
              }).join('&'))
            }, text);
          };
        }

        if (item.displayColumnType === '1' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '0') {
          columnTemp['render'] = function (text, record) {
            return item.displayPropertyList.map(function (itemMap) {
              return _this.columnRenderEnum["".concat(itemMap.type).concat(itemMap.optionDataSource || '')] ? _this.columnRenderEnum["".concat(itemMap.type).concat(itemMap.optionDataSource || '')](record[itemMap.code], record, itemMap.dictName) : record[itemMap.code];
            }).join(item.divider || '');
          };
        }

        if (item.displayColumnType === '1' && (item === null || item === void 0 ? void 0 : item.hyperlink) == '1') {
          columnTemp['render'] = function (text, record) {
            var content = item.displayPropertyList.map(function (itemMap) {
              return _this.columnRenderEnum["".concat(itemMap.type).concat(itemMap.optionDataSource || '')] ? _this.columnRenderEnum["".concat(itemMap.type).concat(itemMap.optionDataSource || '')](record[itemMap.code], record, itemMap.dictName) : record[itemMap.code];
            }).join(item.divider || '');
            return /*#__PURE__*/React.createElement(Link, {
              to: "".concat(item === null || item === void 0 ? void 0 : item.routingAddress, "?").concat(((item === null || item === void 0 ? void 0 : item.hyperlinkParam) || []).map(function (key) {
                return "".concat(key, "=").concat(record[key]);
              }).join('&'))
            }, content);
          };
        }

        if (index < Number(lockColumn)) {
          columnTemp['fixed'] = 'left';
        }

        return columnTemp;
      });
      var formList = arg['queryCondition'].map(function (item) {
        return _this._getFormAttr(item);
      });
      var editParam = Array.isArray(arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig2 = arg.customFromConfig) === null || _arg$customFromConfig2 === void 0 ? void 0 : _arg$customFromConfig2.editParam) ? arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig3 = arg.customFromConfig) === null || _arg$customFromConfig3 === void 0 ? void 0 : _arg$customFromConfig3.editParam.filter(function (item) {
        return item !== null;
      }) : [];
      var editLink = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig4 = arg.customFromConfig) === null || _arg$customFromConfig4 === void 0 ? void 0 : _arg$customFromConfig4.editPage;
      var createLink = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig5 = arg.customFromConfig) === null || _arg$customFromConfig5 === void 0 ? void 0 : _arg$customFromConfig5.createPage;
      var deleteUrl = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig6 = arg.customFromConfig) === null || _arg$customFromConfig6 === void 0 ? void 0 : _arg$customFromConfig6.deleteInterfaceAddress;
      var deleteFetchMethod = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig7 = arg.customFromConfig) === null || _arg$customFromConfig7 === void 0 ? void 0 : _arg$customFromConfig7.deleteHttpMethod;
      var queryBtnUrl = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig8 = arg.customFromConfig) === null || _arg$customFromConfig8 === void 0 ? void 0 : _arg$customFromConfig8.queryInterfaceAddress;
      var queryFetchMethod = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig9 = arg.customFromConfig) === null || _arg$customFromConfig9 === void 0 ? void 0 : _arg$customFromConfig9.queryHttpMethod;
      var isOpenCreate = Boolean(Number(arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig10 = arg.customFromConfig) === null || _arg$customFromConfig10 === void 0 ? void 0 : _arg$customFromConfig10.createDisable));
      var isOpenEdit = Boolean(Number(arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig11 = arg.customFromConfig) === null || _arg$customFromConfig11 === void 0 ? void 0 : _arg$customFromConfig11.editDisable));
      var isOpenQuery = Boolean(Number(arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig12 = arg.customFromConfig) === null || _arg$customFromConfig12 === void 0 ? void 0 : _arg$customFromConfig12.queryDisable));
      var isOpenDelete = Boolean(Number(arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig13 = arg.customFromConfig) === null || _arg$customFromConfig13 === void 0 ? void 0 : _arg$customFromConfig13.deleteDisable));
      var defaultSortFields = [arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig14 = arg.customFromConfig) === null || _arg$customFromConfig14 === void 0 ? void 0 : _arg$customFromConfig14.defaultSortColumn, sortEnum[arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig15 = arg.customFromConfig) === null || _arg$customFromConfig15 === void 0 ? void 0 : _arg$customFromConfig15.sortRule]];
      var sortSplitStr = (arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig16 = arg.customFromConfig) === null || _arg$customFromConfig16 === void 0 ? void 0 : _arg$customFromConfig16.sortSeparator) || ' ';
      var isShowOption = Boolean(Number(arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig17 = arg.customFromConfig) === null || _arg$customFromConfig17 === void 0 ? void 0 : _arg$customFromConfig17.showOperationColumn));
      var queryButtonCode = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig18 = arg.customFromConfig) === null || _arg$customFromConfig18 === void 0 ? void 0 : _arg$customFromConfig18.queryButtonCode;
      var deleteButtonCode = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig19 = arg.customFromConfig) === null || _arg$customFromConfig19 === void 0 ? void 0 : _arg$customFromConfig19.deleteButtonCode;
      var createButtonCode = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig20 = arg.customFromConfig) === null || _arg$customFromConfig20 === void 0 ? void 0 : _arg$customFromConfig20.createButtonCode;
      var editButtonCode = arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig21 = arg.customFromConfig) === null || _arg$customFromConfig21 === void 0 ? void 0 : _arg$customFromConfig21.editButtonCode;
      var optionColumn = {
        title: t('portal_operating'),
        key: 'option',
        dataIndex: 'option',
        fixed: 'right',
        width: 200,
        render: function render(text, record) {
          return /*#__PURE__*/React.createElement(React.Fragment, null, isOpenEdit && _this.power(_this.state.editButtonCode) && /*#__PURE__*/React.createElement(Link, {
            to: "".concat(editLink, "?").concat(editParam.map(function (key) {
              return "".concat(key, "=").concat(record[key]);
            }).join('&'))
          }, t('portal_edit')), isOpenDelete && isOpenEdit && /*#__PURE__*/React.createElement(Divider, {
            type: "vertical"
          }), isOpenDelete && _this.power(_this.state.deleteButtonCode) && /*#__PURE__*/React.createElement(Popconfirm, {
            title: t('portal_delete_tip'),
            placement: "leftBottom",
            icon: /*#__PURE__*/React.createElement(QuestionCircleOutlined, {
              style: {
                color: 'red'
              }
            }),
            onConfirm: function onConfirm() {
              return _this._handleDelete(record.id);
            }
          }, /*#__PURE__*/React.createElement("span", {
            style: {
              cursor: 'pointer',
              color: '#1890ff'
            }
          }, t('portal_delete'))), _this._initSolt('option', record));
        }
      };
      columnList = isShowOption ? [].concat(_toConsumableArray(columnList), [optionColumn]) : _toConsumableArray(columnList);

      var allColumnList = _toConsumableArray(columnList);

      var filterColumnOptionList = allColumnList.map(function (item) {
        return {
          label: item.title,
          value: item.key
        };
      });
      var filterColumnValueList = allColumnList.map(function (item) {
        return item.key;
      });
      var rowSelectionType = rowSelectionTypeEnum[arg === null || arg === void 0 ? void 0 : (_arg$customFromConfig22 = arg.customFromConfig) === null || _arg$customFromConfig22 === void 0 ? void 0 : _arg$customFromConfig22.leftSelectAllBox];
      var stateTemp = {
        columnList: columnList,
        filterColumnOptionList: filterColumnOptionList,
        filterColumnValueList: filterColumnValueList,
        allColumnList: allColumnList,
        formList: formList,
        queryBtnUrl: queryBtnUrl,
        createLink: createLink,
        deleteUrl: deleteUrl,
        isOpenCreate: isOpenCreate,
        isOpenEdit: isOpenEdit,
        isOpenQuery: isOpenQuery,
        isOpenDelete: isOpenDelete,
        defaultSortFields: defaultSortFields,
        rowSelectionType: rowSelectionType,
        queryFetchMethod: queryFetchMethod,
        deleteFetchMethod: deleteFetchMethod,
        sortSplitStr: sortSplitStr,
        isShowOption: isShowOption,
        queryButtonCode: queryButtonCode,
        deleteButtonCode: deleteButtonCode,
        createButtonCode: createButtonCode,
        editButtonCode: editButtonCode
      };

      if ((_this$props = _this.props) !== null && _this$props !== void 0 && (_this$props$initialVa = _this$props.initialValues) !== null && _this$props$initialVa !== void 0 && _this$props$initialVa.pageNum) {
        var _this$props2;

        stateTemp['current'] = Number((_this$props2 = _this.props) === null || _this$props2 === void 0 ? void 0 : _this$props2.pageNum);
      }

      if ((_this$props3 = _this.props) !== null && _this$props3 !== void 0 && (_this$props3$initialV = _this$props3.initialValues) !== null && _this$props3$initialV !== void 0 && _this$props3$initialV.pageSize) {
        var _this$props4;

        stateTemp['pageSize'] = (_this$props4 = _this.props) === null || _this$props4 === void 0 ? void 0 : _this$props4.pageSize;
      }

      if ((_this$props5 = _this.props) !== null && _this$props5 !== void 0 && (_this$props5$initialV = _this$props5.initialValues) !== null && _this$props5$initialV !== void 0 && _this$props5$initialV.sort) {
        var _this$props6;

        stateTemp['defaultSortFields'] = (_this$props6 = _this.props) === null || _this$props6 === void 0 ? void 0 : _this$props6.sort.split(' ');
      }

      _this.setState(stateTemp, function () {
        _this._handleBtnSubmitQuery();
      });
    });

    _defineProperty(_assertThisInitialized(_this), "_getFormAttr", function (itemMode) {
      var _filter$, _filter$2, _itemMode$property;

      var property = {}; // 0：单行文本，1：多行文本，2：数字，3：下拉框，4：日期，5：图片，6：附件，7：关联属性，8：公式计算'
      // property['label'] = itemMode?.name;

      property['name'] = itemMode === null || itemMode === void 0 ? void 0 : itemMode.code;
      property['formType'] = itemMode === null || itemMode === void 0 ? void 0 : itemMode.type;
      property['label'] = _this.props.t(itemMode === null || itemMode === void 0 ? void 0 : itemMode.languageEncoding);
      property['component'] = itemMode === null || itemMode === void 0 ? void 0 : itemMode.component;
      property['columnNum'] = 4;
      property['addonBefore'] = ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.prefixConfig) || []).map(function (item) {
        return {
          label: item.name,
          value: item.name
        };
      });
      property['addonBeforeDefault'] = (_filter$ = ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.prefixConfig) || []).filter(function (item) {
        return item.ifDefault;
      })[0]) === null || _filter$ === void 0 ? void 0 : _filter$.name;
      property['addonAfter'] = ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.suffixConfig) || []).map(function (item) {
        return {
          label: item.name,
          value: item.name
        };
      });
      property['addonAfterDefault'] = (_filter$2 = ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.suffixConfig) || []).filter(function (item) {
        return item.ifDefault;
      })[0]) === null || _filter$2 === void 0 ? void 0 : _filter$2.name;
      (itemMode === null || itemMode === void 0 ? void 0 : (_itemMode$property = itemMode.property) === null || _itemMode$property === void 0 ? void 0 : _itemMode$property.selectorType) === '3' ? property['isMultipleSelect'] = true : property['isMultipleSelect'] = false;
      property['defaultValue'] = (itemMode === null || itemMode === void 0 ? void 0 : itemMode.defaultValue) || undefined;
      property['optionList'] = ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.propertyOptionList) || []).map(function (item) {
        return {
          label: item.label || item.value,
          value: item.value
        };
      });

      if ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.optionDataSource) === '2') {
        property['optionList'] = getDictionaryListByType(itemMode === null || itemMode === void 0 ? void 0 : itemMode.dictName).map(function (item) {
          return {
            label: item.dictLabelCn,
            value: item.dictValue
          };
        });
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.type) === '4') {
        _this.setState({
          formItemCode: [].concat(_toConsumableArray(_this.state.formItemCode), [itemMode.code])
        });
      } // selectorType  1-单选  2-复选框  否则是下拉（0-下拉单选 3-下拉多选）


      if ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.type) === '3' && (itemMode === null || itemMode === void 0 ? void 0 : itemMode.selectorType) === '1') {
        property['formType'] = '3_1';
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.type) === '3' && (itemMode === null || itemMode === void 0 ? void 0 : itemMode.selectorType) === '2') {
        property['formType'] = '3_2';
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.type) === '4' && (itemMode === null || itemMode === void 0 ? void 0 : itemMode.selectorType) === '1') {
        property['formType'] = '4_1';
      }

      if ((itemMode === null || itemMode === void 0 ? void 0 : itemMode.type) === '4' && (itemMode === null || itemMode === void 0 ? void 0 : itemMode.selectorType) === '2') {
        property['isShowTime'] = true;
      }

      return property;
    });

    _defineProperty(_assertThisInitialized(_this), "_getArg", function () {
      var _this$formRef$current, _assertThisInitialize, _assertThisInitialize2;

      var current = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _this.state.current;
      var pageSize = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _this.state.pageSize;
      var fieldOrder = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : _this.state.defaultSortFields;
      var formArg = (_this$formRef$current = _this.formRef.current) === null || _this$formRef$current === void 0 ? void 0 : _this$formRef$current.getFieldsValue();

      var arg = _objectSpread(_objectSpread({}, formArg), {}, {
        pageNum: current,
        pageNo: current,
        pageSize: pageSize,
        sort: !!fieldOrder[0] ? fieldOrder.join((_assertThisInitialize = _assertThisInitialized(_this)) === null || _assertThisInitialize === void 0 ? void 0 : (_assertThisInitialize2 = _assertThisInitialize.state) === null || _assertThisInitialize2 === void 0 ? void 0 : _assertThisInitialize2.sortSplitStr) : null
      });

      _this.state.formItemCode.forEach(function (key) {
        if (!arg[key]) return;
        arg[key] = moment(arg[key]).format('YYYY-MM-DD HH:mm:ss');
      });

      return arg;
    });

    _defineProperty(_assertThisInitialized(_this), "_handleFetchList", /*#__PURE__*/function () {
      var _ref2 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee2(current, pageSize, fieldOrder) {
        var arg, _res$data, _res$data2, res;

        return _regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _this.setState({
                  isLoading: true
                });

                arg = _this._getArg(current, pageSize, fieldOrder);
                _context2.prev = 2;
                _context2.next = 5;
                return fetch[_this.state.queryFetchMethod](_this.state.queryBtnUrl, arg, '/api/inner');

              case 5:
                res = _context2.sent;

                if (typeof _this.props.onChange === 'function') {
                  _this.props.onChange(arg);
                }

                (res === null || res === void 0 ? void 0 : (_res$data = res.data) === null || _res$data === void 0 ? void 0 : _res$data.content) && _this.setState({
                  tableHeight: getTableHeight(_this.state.tableClassName, 171),
                  sourceList: res.data.content,
                  isPagination: true,
                  pageSize: res.data.pageSize,
                  current: res.data.pageNo || res.data.pageNum,
                  total: res.data.total || res.data.totalElements,
                  isLoading: false
                });
                !(res !== null && res !== void 0 && (_res$data2 = res.data) !== null && _res$data2 !== void 0 && _res$data2.content) && _this.setState({
                  tableHeight: getTableHeight(_this.state.tableClassName, 171 - 54),
                  sourceList: res.data,
                  isPagination: false,
                  isLoading: false
                });
                _context2.next = 15;
                break;

              case 11:
                _context2.prev = 11;
                _context2.t0 = _context2["catch"](2);

                _this.setState({
                  isLoading: false
                });

                message.error('MEK列表请求失败，请检查查询API是否正确无误');

              case 15:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2, null, [[2, 11]]);
      }));

      return function (_x, _x2, _x3) {
        return _ref2.apply(this, arguments);
      };
    }());

    _defineProperty(_assertThisInitialized(_this), "_handleBtnSubmitQuery", function () {
      _this._handleFetchList(1);
    });

    _defineProperty(_assertThisInitialized(_this), "_handleOnChange", function (current, pageSize, fieldOrder) {
      _this.setState({
        current: current,
        pageSize: pageSize,
        fieldOrder: fieldOrder
      });

      _this._handleFetchList(current, pageSize, fieldOrder);
    });

    _defineProperty(_assertThisInitialized(_this), "_handleResetFields", function () {
      var _this$formRef$current2;

      (_this$formRef$current2 = _this.formRef.current) === null || _this$formRef$current2 === void 0 ? void 0 : _this$formRef$current2.resetFields();
    });

    _defineProperty(_assertThisInitialized(_this), "_handleDelete", /*#__PURE__*/function () {
      var _ref3 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee3(arg) {
        var res;
        return _regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _context3.prev = 0;
                _context3.next = 3;
                return fetch[_this.state.deleteFetchMethod](_this.state.deleteUrl, {
                  id: arg
                }, '/api/inner');

              case 3:
                res = _context3.sent;

                _this._handleFetchList();

                _context3.next = 10;
                break;

              case 7:
                _context3.prev = 7;
                _context3.t0 = _context3["catch"](0);
                message.error('删除失败');

              case 10:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, null, [[0, 7]]);
      }));

      return function (_x4) {
        return _ref3.apply(this, arguments);
      };
    }());

    _defineProperty(_assertThisInitialized(_this), "_handleShow", function () {
      _this.setState({
        isShow: !_this.state.isShow
      }, function () {
        _this.setState({
          tableHeight: getTableHeight(_this.state.tableClassName)
        });
      });
    });

    _defineProperty(_assertThisInitialized(_this), "_onChangeFormList", function (formList) {
      var formListTemp = _toConsumableArray(formList);

      if (formList.length > 4) {
        formListTemp.length = 4;
      }

      return formListTemp;
    });

    _defineProperty(_assertThisInitialized(_this), "_handleCreateLink", function () {
      _this.props.history.push(_this.state.createLink);
    });

    _defineProperty(_assertThisInitialized(_this), "_initSolt", function (soltSite, record) {
      return React.Children.map(_this.props.children, function (child, i) {
        if (child.props.solt === soltSite) {
          var _child$props;

          if (child !== null && child !== void 0 && (_child$props = child.props) !== null && _child$props !== void 0 && _child$props.render) {
            return _this._getSoltArg(child.props.render(record), record, soltSite);
          }

          return _this._getSoltArg(child.props.children, record, soltSite);
        }

        return '';
      });
    });

    _defineProperty(_assertThisInitialized(_this), "_handleChangeColumn", function (props) {
      var columnList = _this.state.allColumnList.filter(function (item) {
        return props.includes(item.key);
      });

      _this.setState({
        filterColumnValueList: _toConsumableArray(props),
        columnList: columnList
      });
    });

    _defineProperty(_assertThisInitialized(_this), "_handleResetCheckGroup", function () {
      var filterColumnValueList = _this.state.allColumnList.map(function (item) {
        return item.key;
      });

      _this._handleChangeColumn(filterColumnValueList);
    });

    _defineProperty(_assertThisInitialized(_this), "rowSelection", {
      type: _this.state.rowSelectionType,
      onChange: function onChange(selectedRowKeys, selectedRows) {
        _this.setState({
          selectedRowKeys: selectedRowKeys
        });
      }
    });

    _defineProperty(_assertThisInitialized(_this), "_getPagination", function () {
      return {
        showQuickJumper: true,
        showSizeChanger: true,
        defaultPageSize: 50,
        position: ['bottomLeft'],
        current: _this.state.current,
        pageSize: _this.state.pageSize,
        total: _this.state.total,
        showTotal: function showTotal(total) {
          return "".concat(_this.props.t('portal_total'), " ").concat(total, " ").concat(_this.props.t('portal_item'));
        }
      };
    });

    _defineProperty(_assertThisInitialized(_this), "power", function (code) {
      if (!code) return true;
      var res = JSON.parse(window.localStorage.getItem('MYPERMISSION') || []);
      return res.includes(code);
    });

    return _this;
  }

  _createClass(DynamicTable, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (this.props.modelCode !== prevProps.modelCode || this.props.formCode !== prevProps.formCode) {
        this.initData();
      }
    } // 向后端请求数据

  }, {
    key: "_getSoltArg",
    value: function _getSoltArg(children, record, soltSite) {
      var _that = this;

      return React.Children.map(children, function (child, i) {
        var _child$props2;

        if (typeof (child === null || child === void 0 ? void 0 : (_child$props2 = child.props) === null || _child$props2 === void 0 ? void 0 : _child$props2.onClick) === 'function') {
          return /*#__PURE__*/React.cloneElement(child, {
            onClick: function onClick(e) {
              return child.props.onClick(e, soltSite === 'header' ? _that.formRef.current.getFieldsValue() : record);
            }
          });
        }

        return child;
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var props = this.props;
      var t = this.props.t;
      var _this$state = this.state,
          columnList = _this$state.columnList,
          formList = _this$state.formList,
          sourceList = _this$state.sourceList,
          tableClassName = _this$state.tableClassName,
          tableHeight = _this$state.tableHeight;
      var formItemList = formList.length <= 4 ? formList : this.state.isShow ? _toConsumableArray(formList) : this._onChangeFormList(formList);
      return /*#__PURE__*/React.createElement("div", {
        className: tableClassName
      }, formItemList.length != 0 && /*#__PURE__*/React.createElement(Form, {
        layout: "vertical",
        size: "middle",
        ref: this.formRef,
        className: "search-area",
        initialValues: this.props.initialValues
      }, /*#__PURE__*/React.createElement(Row, {
        gutter: 30
      }, formItemList.map(function (item, index) {
        return formItem["formItem".concat(item.formType)](_objectSpread({}, item));
      })), /*#__PURE__*/React.createElement(Row, {
        gutter: 30
      }, /*#__PURE__*/React.createElement(Col, {
        span: 24
      }, this.state.isOpenQuery && this.power(this.state.queryButtonCode) && /*#__PURE__*/React.createElement(Button, {
        type: "primary",
        onClick: this._handleBtnSubmitQuery
      }, t('portal_search')), this.state.isOpenQuery && /*#__PURE__*/React.createElement(Button, {
        onClick: this._handleResetFields
      }, t('portal_reset')), formList.length > 4 && /*#__PURE__*/React.createElement(Button, {
        type: "link",
        size: "small",
        onClick: this._handleShow
      }, this.state.isShow ? t('hide_query_criteria') : t('portal_query_criteria'), this.state.isShow ? /*#__PURE__*/React.createElement(UpOutlined, null) : /*#__PURE__*/React.createElement(DownOutlined, null))))), /*#__PURE__*/React.createElement("div", {
        className: "mek-table-top",
        style: {
          padding: '14px 14px 0 14px',
          paddingBottom: '10px',
          background: '#fff',
          overflow: 'hidden'
        }
      }, this.state.isOpenCreate && this.power(this.state.createButtonCode) && /*#__PURE__*/React.createElement(Button, {
        type: "primary",
        onClick: this._handleCreateLink
      }, t('portal_add')), this._initSolt('header'), /*#__PURE__*/React.createElement(Button, {
        onClick: this._handleResetCheckGroup,
        style: {
          "float": 'right',
          marginTop: '-2px'
        },
        icon: /*#__PURE__*/React.createElement(ReloadOutlined, null),
        type: "link",
        size: "small"
      }, t('portal_reset')), /*#__PURE__*/React.createElement(Checkbox.Group, {
        style: {
          "float": 'right'
        },
        options: this.state.filterColumnOptionList,
        value: this.state.filterColumnValueList,
        onChange: this._handleChangeColumn
      })), /*#__PURE__*/React.createElement(Table, {
        className: "mek-table-content",
        columns: columnList,
        onChange: function onChange(pagination, b, fieldOrder) {
          _this2._handleOnChange(pagination.current, pagination.pageSize, fieldOrder.field ? [fieldOrder.field, sortEnum[fieldOrder.order]] : undefined);
        },
        dataSource: sourceList,
        rowKey: function rowKey(record) {
          var _record$children;

          return (_record$children = record.children) !== null && _record$children !== void 0 && _record$children.length ? record.id : record.id + Math.random();
        },
        loading: this.state.isLoading,
        rowSelection: this.rowSelectionType ? this.rowSelection : undefined,
        pagination: this.state.isPagination ? this._getPagination() : false,
        scroll: {
          x: '100%'
        },
        style: {
          padding: '0 14px 14px 14px',
          background: '#fff'
        }
      }));
    }
  }]);

  return DynamicTable;
}(PureComponent);

DynamicTable.defaultProps = {
  modelCode: '',
  formCode: ''
};
export default withTranslation()(withRouter(DynamicTable));
