// @mcd/portal-components/dist/dynamic/dynamicItem/formItem.js
import _extends from "@babel/runtime/helpers/extends";

/**
 * @param  {Object} props
 * @return {Object and Function}
 * @author ethan.yan
 */
import React, { useState, Component } from 'react';
import { Input, InputNumber, Select, DatePicker, TimePicker, Col, Radio, Checkbox, Form } from '../../index';
import moment from 'moment';
import { Upload } from './upload';
import * as CusComponents from '../../index';

var rulePattern = function rulePattern(validPattern, validMessage) {
  // 单行文本框的校验，0：无校验，1：自定义校验，2：手机号校验，3：邮箱校验，4：身份证校验
  return {
    '1': [validPattern, validMessage],
    '2': ['^((17[0-9])|(14[0-9])|(13[0-9])|(15[^4,\\D])|(18[0,0-9]))\\d{8}$', validMessage],
    '3': ['^([a-zA-Z0-9_\\-\\.]+)@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.)|(([a-zA-Z0-9\\-]+\\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\\]?)$', validMessage],
    '4': ['^[1-9]\\d{5}[1-9]\\d{3}((0\\d)|(1[0-2]))(([0|1|2]\\d)|3[0-1])\\d{3}([0-9]|X)$', validMessage]
  };
};

var handleSolt = function handleSolt(children, code) {
  return React.Children.map(children, function (child, i) {
    if (child.props.solt === code) {
      return child.props.children;
    }

    return '';
  });
};

var hasSoltRowOrColumn = function hasSoltRowOrColumn(props, children) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum * props.soltColumn,
    key: props.name
  }, handleSolt(children, props.name));
};

var noSoltRowOrColumn = function noSoltRowOrColumn(props, children) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, handleSolt(children, props.name));
};

export var formItem0 = function formItem0(props) {
  var validPattern = rulePattern(props === null || props === void 0 ? void 0 : props.validPattern, props === null || props === void 0 ? void 0 : props.validMessage)[props === null || props === void 0 ? void 0 : props.validType] || [];
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    validateTrigger: "onBlur",
    rules: [{
      required: Boolean(Number(props.required)),
      message: "".concat(props.label)
    }, {
      pattern: validPattern[0],
      message: validPattern[1]
    }]
  }, /*#__PURE__*/React.createElement(Input, {
    placeholder: "\u8BF7\u8F93\u5165".concat(props.label),
    disabled: props.disabled,
    style: {
      width: '100%'
    },
    maxLength: props === null || props === void 0 ? void 0 : props.maxLength,
    addonBefore: props.addonBefore.length > 0 && /*#__PURE__*/React.createElement(Form.Item, {
      name: "".concat(props.name, "_prefix"),
      key: props.name,
      initialValue: props.addonBeforeDefault,
      noStyle: true
    }, /*#__PURE__*/React.createElement(Select, {
      placeholder: "\u8BF7\u9009\u62E9",
      allowClear: true,
      disabled: props.disabled,
      style: {
        minWidth: '70px'
      },
      getPopupContainer: function getPopupContainer(node) {
        return node.parentNode;
      }
    }, props.addonBefore.map(function (item) {
      return /*#__PURE__*/React.createElement(Select.Option, {
        key: item.value,
        value: item.value
      }, item.label);
    }))),
    addonAfter: props.addonAfter.length > 0 && /*#__PURE__*/React.createElement(Form.Item, {
      name: "".concat(props.name, "_suffix"),
      key: props.name,
      initialValue: props.addonAfterDefault,
      noStyle: true
    }, /*#__PURE__*/React.createElement(Select, {
      placeholder: "\u8BF7\u9009\u62E9",
      allowClear: true,
      disabled: props.disabled,
      style: {
        minWidth: '70px'
      },
      getPopupContainer: function getPopupContainer(node) {
        return node.parentNode;
      }
    }, props.addonAfter.map(function (item) {
      return /*#__PURE__*/React.createElement(Select.Option, {
        key: item.value,
        value: item.value
      }, item.label);
    }))),
    allowClear: true
  })));
};
export var formItem1 = function formItem1(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    validateTrigger: "onBlur",
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(Input.TextArea, {
    placeholder: "\u8BF7\u8F93\u5165".concat(props.label),
    disabled: props.disabled,
    style: {
      width: '100%'
    },
    maxLength: props === null || props === void 0 ? void 0 : props.maxLength
  })));
};
export var formItem2 = function formItem2(props) {
  var widthStyle = '100%';

  if (props.addonBefore.length > 0 && props.addonAfter.length > 0) {
    widthStyle = 'calc(100% - 140px)';
  }

  if (props.addonBefore.length > 0 && props.addonAfter.length !== 0 || props.addonBefore.length === 0 && props.addonAfter.length > 0) {
    widthStyle = 'calc(100% - 70px)';
  }

  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    style: {
      fontSize: '12px'
    },
    labelCol: {
      fontSize: '12px'
    },
    required: Boolean(Number(props.required))
  }, /*#__PURE__*/React.createElement(Input.Group, {
    compact: true
  }, props.addonBefore.length > 0 && /*#__PURE__*/React.createElement(Form.Item, {
    name: "".concat(props.name, "_prefix"),
    initialValue: props.addonBeforeDefault,
    noStyle: true
  }, /*#__PURE__*/React.createElement(Select, {
    placeholder: "\u8BF7\u9009\u62E9}",
    allowClear: true,
    disabled: props.disabled,
    style: {
      minWidth: '70px'
    },
    getPopupContainer: function getPopupContainer(node) {
      return node.parentNode;
    }
  }, props.addonBefore.map(function (item) {
    return /*#__PURE__*/React.createElement(Select.Option, {
      key: item.value,
      value: item.value
    }, item.label);
  }))), /*#__PURE__*/React.createElement(Form.Item, {
    name: props.name,
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    validateTrigger: "onBlur",
    noStyle: true,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(InputNumber, {
    placeholder: "\u8BF7\u8F93\u5165".concat(props.label),
    disabled: props.disabled,
    style: {
      minWidth: widthStyle
    },
    max: Number.parseInt(props.maxValue) || Number.MAX_SAFE_INTEGER,
    min: Number.parseInt(props.minValue) || Number.MIN_SAFE_INTEGER,
    precision: props.precision || undefined,
    maxLength: props === null || props === void 0 ? void 0 : props.maxLength
  })), props.addonAfter.length > 0 && /*#__PURE__*/React.createElement(Form.Item, {
    name: "".concat(props.name, "_suffix"),
    initialValue: props.addonAfterDefault,
    noStyle: true
  }, /*#__PURE__*/React.createElement(Select, {
    placeholder: "\u8BF7\u9009\u62E9",
    allowClear: true,
    disabled: props.disabled,
    style: {
      minWidth: '70px'
    },
    getPopupContainer: function getPopupContainer(node) {
      return node.parentNode;
    }
  }, props.addonAfter.map(function (item) {
    return /*#__PURE__*/React.createElement(Select.Option, {
      key: item.value,
      value: item.value
    }, item.label);
  }))))));
};
export var formItem3_1 = function formItem3_1(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u9009\u62E9".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(Radio.Group, null, props.optionList.map(function (item) {
    return /*#__PURE__*/React.createElement(Radio, {
      key: item.value,
      value: item.value
    }, item.label);
  }))));
};
export var formItem3_2 = function formItem3_2(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u9009\u62E9".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(Checkbox.Group, {
    options: props.optionList,
    disabled: props.disabled
  })));
};
export var formItem3 = function formItem3(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: props !== null && props !== void 0 && props.isMultipleSelect ? 24 : 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(Select, {
    placeholder: "\u8BF7\u9009\u62E9".concat(props.label),
    allowClear: true,
    disabled: props.disabled,
    style: {
      width: '100%'
    },
    getPopupContainer: function getPopupContainer(node) {
      return node.parentNode;
    },
    mode: props !== null && props !== void 0 && props.isMultipleSelect ? 'multiple' : undefined
  }, props.optionList.map(function (item) {
    var value = item.value; // store页面特殊处理  

    if (item.value === 'true') {
      value = true;
    }

    if (item.value === 'false') {
      value = false;
    }

    return /*#__PURE__*/React.createElement(Select.Option, {
      key: item.value,
      value: value
    }, item.label);
  }))));
};
export var formItem4_1 = function formItem4_1(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(TimePicker, {
    placeholder: "\u8BF7\u9009\u62E9".concat(props.label),
    allowClear: true,
    disabled: props.disabled,
    style: {
      width: '100%'
    },
    format: (props === null || props === void 0 ? void 0 : props.dateFormat) || 'HH:mm:ss',
    getPopupContainer: function getPopupContainer(node) {
      return node.parentNode;
    }
  })));
};
export var formItem4 = function formItem4(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue ? moment(props.defaultValue) : '',
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(DatePicker, {
    placeholder: "\u8BF7\u9009\u62E9".concat(props.label),
    allowClear: true,
    disabled: props.disabled,
    style: {
      width: '100%'
    },
    format: (props === null || props === void 0 ? void 0 : props.dateFormat) || 'YYYY-MM-DD HH:mm:ss',
    getPopupContainer: function getPopupContainer(node) {
      return node.parentNode;
    },
    showTime: props !== null && props !== void 0 && props.isShowTime ? {
      format: 'HH:mm:ss'
    } : false
  })));
};
export var formItem5 = function formItem5(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(Upload, _extends({}, props, {
    listType: "picture-card"
  }))));
};
export var formItem6 = function formItem6(props) {
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    valuePropName: "fileList",
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(Upload, _extends({}, props, {
    listType: "text"
  }))));
};
export var formItemslot = function formItemslot(props, children) {
  if (!props.soltRow || !props.soltColumn) {
    return noSoltRowOrColumn(props, children);
  }

  return hasSoltRowOrColumn(props, children);
};
export var formItem12 = function formItem12(props) {
  if (!props.component || !CusComponents[props.component]) {
    return '';
  }

  var CusComponent = CusComponents[props.component];
  return /*#__PURE__*/React.createElement(Col, {
    span: 24 / props.columnNum,
    key: props.name
  }, /*#__PURE__*/React.createElement(Form.Item, {
    label: "".concat(props.label, "\uFF1A"),
    name: props.name,
    style: {
      fontSize: '12px'
    },
    key: props.name,
    extra: props === null || props === void 0 ? void 0 : props.extra,
    initialValue: props.defaultValue,
    rules: [{
      required: Boolean(Number(props.required)),
      message: "\u8BF7\u8F93\u5165".concat(props.label)
    }]
  }, /*#__PURE__*/React.createElement(CusComponent, null)));
};
/**
 * 0：单行文本，1：多行文本，2：数字，3：下拉框，3_1: 单选， 3_2：多选
 * 4：日期，4_1： 时间， 5：图片， 6：附件，7：关联属性，8：公式计算'
 * solt: 插槽 12: 自定义组件
 */
