// @mcd/boss-common/dist/utils/cache.js
// key
export var Constants = {
  SSO: 'SSO',
  getSSOInfo: 'getSSOInfo',
  Error: 'Error',
  LOADING_NUM: 'LOADING_NUM',
  TITLE: 'TITLE',
  SHOW_SIDEBAR: 'SHOW_SIDEBAR',
  STORAGE_VERSION: 'STORAGE_VERSION',
  USER_INFO: 'USER_INFO',
  DICTIONARY: 'DICTIONARY',
  DICTIONARYLIST: 'DICTIONARYLIST',
  PARAMETERS: 'PARAMETERS',
  LANGSWITCH: 'LANGSWITCH',
  THEMESWITCH: 'THEMESWITCH',
  MENU: 'MENU',
  I18N_DATA: 'I18N_DATA',
  LOCALE: 'locale',
  Theme_Color: 'Theme_Color',
  CLIENT_ROUTERS: 'CLIENT_ROUTERS',
  LEFT_MENU: 'LEFT_MENU',
  CURRENT_MENUID: 'CURRENT_MENUID',
  CURRENT_MENUNAME: 'CURRENT_MENUNAME',
  CURRENT_PATH: 'CURRENT_PATH',
  COMPANY: 'COMPANY',
  LANG_TYPE: 'LANG_TYPE',
  ISLOADING: 'ISLOADING',
  SSO_HREF: 'SSO_HREF',
  ROUTER: 'ROUTER',
  MYPERMISSION: 'MYPERMISSION',
  PATH_VALUE: 'PATH_VALUE',
  APPLICATION: 'APPLICATION',
  MENU_NAME_EN_CN: 'MENU_NAME_EN_CN',
  CURRENT_APPNAME: 'CURRENT_APPNAME'
}; // 获取缓存

export function getLocalItem(name) {
  if (!global.window || !name) {
    return;
  }

  var content = window.localStorage.getItem(name);

  var _expire = window.localStorage.getItem("".concat(name, "_expire"));

  if (_expire) {
    var now = parseInt(new Date().getTime() / 1000);

    if (now > _expire) {
      return;
    }
  }

  try {
    return JSON.parse(content);
  } catch (e) {
    return content;
  }
} // 设置缓存

export function setLocalItem(name, content) {
  var maxAge = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : null;

  if (!global.window || !name) {
    return;
  }

  if (typeof content !== 'string') {
    content = JSON.stringify(content);
  }

  var storage = global.window.localStorage;
  storage.setItem(name, content);

  if (maxAge && !isNaN(parseInt(maxAge))) {
    var timeout = parseInt(new Date().getTime() / 1000);
    storage.setItem("".concat(name, "_expire"), timeout + maxAge);
  }
} // 移除缓存

export function removeLocalItem(name) {
  if (!global.window || !name) {
    return;
  }

  window.localStorage.removeItem(name);
  window.localStorage.removeItem("".concat(name, "_expire"));
} // 清除全部

export var clearAll = function clearAll() {
  if (!global.window || !name) {
    return;
  }

  window.localStorage.clear();
};
