// @aurum/pfe-ui/es/upload/MultipleFileUpload.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { ConfigProvider, Upload } from 'antd';
import IconFont from '../common/IconFont';
var Dragger = Upload.Dragger;
export default (function (props) {
  var _React$useContext = React.useContext(ConfigProvider.ConfigContext),
      getPrefixCls = _React$useContext.getPrefixCls;

  var hintText = props.hintText;
  return /*#__PURE__*/React.createElement("div", {
    className: "".concat(getPrefixCls(), "-upload-multiple-file")
  }, /*#__PURE__*/React.createElement(Dragger, _extends({
    progress: {
      strokeColor: '#ffbc0d',
      strokeWidth: 6
    },
    multiple: true,
    listType: "picture",
    showUploadList: {
      showRemoveIcon: true,
      removeIcon: /*#__PURE__*/React.createElement(IconFont, {
        type: "icon-qingchu",
        className: "".concat(getPrefixCls(), "-upload-multiple-file-delete-icon")
      })
    }
  }, props), /*#__PURE__*/React.createElement("div", {
    className: "".concat(getPrefixCls(), "-upload-multiple-file-drag-icon")
  }, /*#__PURE__*/React.createElement(IconFont, {
    type: "icon-shangchuan",
    className: "".concat(getPrefixCls(), "-upload-multiple-file-shangchuan-m")
  })), /*#__PURE__*/React.createElement("div", {
    className: "".concat(getPrefixCls(), "-upload-multiple-file-upload-text")
  }, "\u70B9\u51FB\u6216\u62D6\u62FD\u6587\u4EF6\u5230\u6B64\u5904\u4E0A\u4F20"), /*#__PURE__*/React.createElement("div", {
    className: "".concat(getPrefixCls(), "-upload-multiple-file-upload-hint")
  }, hintText)));
});
